<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="John Powers" />

<meta name="date" content="2025-07-21" />

<title>Schiedea pollen competition</title>

<script src="libs/header-attrs-2.29/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<script src="libs/navigation-1.1/codefolding.js"></script>
<link href="libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore"><em>Schiedea</em> pollen competition</h1>
<h4 class="author"><a href="http://sites.uci.edu/powers/">John
Powers</a></h4>
<h4 class="date">2025-07-21</h4>

</div>


<pre class="r"><code>library(tidyverse)
library(glmmTMB)
library(broom)
library(emmeans)
library(knitr)
library(ape)
library(RColorBrewer)
#devtools::install_github(&#39;gastonstat/arcdiagram&#39;)
library(igraph)
library(arcdiagram)
knitr::opts_chunk$set(comment=&quot;&quot;, cache=T, warning = F, message = F,
                      fig.path = &quot;figures-comp/&quot;, dev = &quot;svglite&quot;, dev.args=list(fix_text_size=FALSE))
options(digits=4, knitr.kable.NA = &quot;&quot;)
ggplot2::theme_set(theme_minimal())</code></pre>
<div id="inventory" class="section level1">
<h1>Inventory</h1>
<pre class="r"><code>#in order of phylogenetic distance, pair order is by style length
sp_pairs &lt;- c(&quot;menz&quot;=&quot;menziesii&quot;, &quot;sal&quot;=&quot;salicaria&quot;,
              &quot;lydg&quot;=&quot;lydgatei&quot;, &quot;sarm&quot; =&quot;sarmentosa&quot;,
              &quot;stell&quot;=&quot;stellarioides&quot;,&quot;sperg&quot;=&quot;spergulina&quot;,
              &quot;memb&quot;=&quot;membranacea&quot;,&quot;kaua&quot;=&quot;kauaiensis&quot;,
              &quot;kaal&quot;=&quot;kaalae&quot;, &quot;hook&quot;=&quot;hookeri&quot;)
sp_pal &lt;- set_names(brewer.pal(length(sp_pairs), &quot;Paired&quot;), sp_pairs)
sp_pal_dark &lt;- set_names(sp_pal[rep(seq(2,10,by=2), each=2)], sp_pairs)
typenames &lt;- set_names(c(&quot;Intraspecific&quot;,&quot;Competitive&quot;,&quot;Interspecific&quot;),c(&quot;intra&quot;,&quot;mixed&quot;,&quot;inter&quot;))
type_pal &lt;- set_names(brewer.pal(6,&quot;Pastel1&quot;)[c(2,3,5)], names(typenames))
type_pal_dark &lt;- set_names(brewer.pal(6,&quot;Set1&quot;)[c(2,3,5)], names(typenames))
propnames &lt;- set_names(c(&quot;Capsule formation&quot;,&quot;Seed number&quot;,&quot;Germination&quot;,&quot;Survival&quot;,&quot;Hybrid&quot;), c(&quot;capsule.formed&quot;,&quot;sds&quot;,&quot;prop.germ&quot;, &quot;prop.alive&quot;, &quot;prop.hybrid&quot;))
vigornames &lt;- c(hybrid.vigor.germsurv=&quot;Hybrid/intraspecific fitness:\ngermination and survival&quot;, 
                hybrid.vigor.all=&quot;Hybrid/intraspecific fitness:\nseeds, germination, survivial&quot;)

momid.fixes &lt;- read_tsv(&quot;data/Pollen competition - momid_fixes.tsv&quot;)
momid.fixes.seeds &lt;- momid.fixes %&gt;% drop_na(momid_fixed) %&gt;% select(momid, momid_fixed) %&gt;% deframe()
momid.fixes.germ &lt;- momid.fixes %&gt;% drop_na(germid, momid_fixed) %&gt;% select(momid, momid_fixed) %&gt;% deframe()
momid.fixes.surv &lt;- momid.fixes %&gt;% drop_na(survid, momid_fixed) %&gt;% select(momid, momid_fixed) %&gt;% deframe()

seeds &lt;- read_tsv(&quot;data/Pollen competition - seeds.tsv&quot;) %&gt;% 
  mutate(dadspm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp1, dadsp2), #rearrange dad1/dad2 to have dadsp match momsp
         dadidm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid1, dadid2),
         dadsp  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp2, dadsp1),
         dadid  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid2, dadid1), .after=momid) %&gt;% 
  select(-matches(&quot;dad(sp|id)[12]&quot;)) %&gt;% 
  mutate(across(matches(&quot;sp&quot;), ~ recode(.x, !!!sp_pairs)),
         across(matches(&quot;(dad|mom)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)), 
         across(matches(&quot;(dad|mom)id&quot;), ~recode(.x, !!!momid.fixes.seeds)),
         sxc = paste(momsp, type, sep=&quot;.&quot;),
         type=factor(type, levels= names(typenames)), 
         across(matches(&quot;sp&quot;), ~ factor(.x, levels = sp_pairs)),
         capsule.formed = sds&gt;0,
         sds.nz = sds+1)

seeds.bycross &lt;- seeds %&gt;% group_by(type, momsp, sxc, momid, dadspm, dadidm, dadsp, dadid) %&gt;% 
  summarize(capsule.formed=mean(capsule.formed), .groups=&quot;drop&quot;)

germ.pot &lt;- read_tsv(&quot;data/Pollen competition - germination.tsv&quot;) %&gt;%  
  mutate(dadspm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp1, dadsp2), #rearrange dad1/dad2 to have dadsp match momsp
         dadidm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid1, dadid2),
         dadsp  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp2, dadsp1),
         dadid  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid2, dadid1), .after=momid) %&gt;% 
  select(-starts_with(&quot;germinated_n&quot;), -transplanted_n, -matches(&quot;dad(sp|id)[12]&quot;)) %&gt;% 
  drop_na(germinated) %&gt;% #exclude one contaminated cross and &quot;intra inv&quot; inviability tests
  mutate(across(matches(&quot;sp&quot;), ~ recode(.x, !!!sp_pairs)),
         across(matches(&quot;(dad|mom)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)), 
         across(matches(&quot;(dad|mom)id&quot;), ~recode(.x, !!!momid.fixes.germ)), #TODO are the momid fixes applicable to dads too?
         sxc = paste(momsp, type, sep=&quot;.&quot;),
         type=factor(type, levels= names(typenames)), 
         across(matches(&quot;sp&quot;), ~ factor(.x, levels = sp_pairs)),
         germinated = ifelse(germinated &gt; planted, planted, germinated))

germ &lt;- germ.pot %&gt;% group_by(date_planted, type, momsp, sxc, momid, dadspm, dadidm, dadsp, dadid) %&gt;% #date_planted splits 30 plant x plant crosses
  summarize(n.pdishes = n(), planted = sum(planted), germinated = sum(germinated), .groups=&quot;drop&quot;) %&gt;% # add together all the pots of one cross
  mutate(prop.germ = germinated/planted, # proportion of planted seeds that germinated 
         prop.germ.nz = (germinated+1)/(planted+1)) 

surv &lt;- read_tsv(&quot;data/Pollen competition - survival.tsv&quot;) %&gt;% 
  mutate(dadspm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp1, dadsp2), #rearrange dad1/dad2 to have dadsp match momsp
         dadidm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid1, dadid2),
         dadsp  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp2, dadsp1),
         dadid  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid2, dadid1), .after=momid) %&gt;% 
  select(-matches(&quot;dad(sp|id)[12]&quot;)) %&gt;% drop_na(dead) %&gt;% #one row excluded
  mutate(across(matches(&quot;(mom|dad)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)), 
         across(matches(&quot;(dad|mom)id&quot;), ~recode(.x, !!!momid.fixes.surv)),
         sxc = paste(momsp, type, sep=&quot;.&quot;),
         type=factor(type, levels= names(typenames)),
         across(matches(&quot;sp&quot;), ~ factor(.x, levels = sp_pairs)),
         across(all_of(c(&quot;sameasmom&quot;, &quot;hybrid&quot;)), ~replace_na(.x,0)), 
         alive = sameasmom + hybrid,
         total = alive + dead, 
         prop.hybrid = hybrid / alive,
         prop.alive = alive / total,
         prop.alive.nz = (alive+1)/(total+1))

germ %&gt;% summarize(across(c(n.pdishes, planted, germinated), sum))</code></pre>
<pre><code># A tibble: 1 Ã— 3
  n.pdishes planted germinated
      &lt;int&gt;   &lt;dbl&gt;      &lt;dbl&gt;
1      1658   27812      18732</code></pre>
<pre class="r"><code>inventory &lt;- surv %&gt;% count(momsp, type, dadsp, name=&quot;crosses_surv&quot;) %&gt;% 
  full_join(seeds %&gt;% count(momsp, type, dadsp, name=&quot;crosses_seeds&quot;)) %&gt;% 
  full_join(germ %&gt;% count(momsp, type, dadsp, name=&quot;crosses_germ&quot;)) %&gt;% 
  full_join(germ.pot %&gt;% count(momsp, type, dadsp, name = &quot;pots_germ&quot;)) %&gt;%
  mutate(pots_per_cross = pots_germ/crosses_germ) 
kable(inventory, caption=&quot;Number of crosses between individual plants used for seeds, germination and survival,
        and how many pots the seeds filled&quot;)</code></pre>
<table style="width:100%;">
<caption>Number of crosses between individual plants used for seeds,
germination and survival, and how many pots the seeds filled</caption>
<colgroup>
<col width="14%" />
<col width="6%" />
<col width="14%" />
<col width="13%" />
<col width="14%" />
<col width="13%" />
<col width="10%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="left">type</th>
<th align="left">dadsp</th>
<th align="right">crosses_surv</th>
<th align="right">crosses_seeds</th>
<th align="right">crosses_germ</th>
<th align="right">pots_germ</th>
<th align="right">pots_per_cross</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">intra</td>
<td align="left">menziesii</td>
<td align="right">7</td>
<td align="right">64</td>
<td align="right">8</td>
<td align="right">74</td>
<td align="right">9.250</td>
</tr>
<tr class="even">
<td align="left">menziesii</td>
<td align="left">mixed</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">170</td>
<td align="right">16</td>
<td align="right">80</td>
<td align="right">5.000</td>
</tr>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">inter</td>
<td align="left">salicaria</td>
<td align="right">19</td>
<td align="right">301</td>
<td align="right">18</td>
<td align="right">87</td>
<td align="right">4.833</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">intra</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">82</td>
<td align="right">11</td>
<td align="right">97</td>
<td align="right">8.818</td>
</tr>
<tr class="odd">
<td align="left">salicaria</td>
<td align="left">mixed</td>
<td align="left">menziesii</td>
<td align="right">14</td>
<td align="right">99</td>
<td align="right">12</td>
<td align="right">76</td>
<td align="right">6.333</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">inter</td>
<td align="left">menziesii</td>
<td align="right">13</td>
<td align="right">125</td>
<td align="right">12</td>
<td align="right">53</td>
<td align="right">4.417</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">intra</td>
<td align="left">lydgatei</td>
<td align="right">5</td>
<td align="right">111</td>
<td align="right">5</td>
<td align="right">42</td>
<td align="right">8.400</td>
</tr>
<tr class="even">
<td align="left">lydgatei</td>
<td align="left">mixed</td>
<td align="left">sarmentosa</td>
<td align="right">5</td>
<td align="right">96</td>
<td align="right">11</td>
<td align="right">56</td>
<td align="right">5.091</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">inter</td>
<td align="left">sarmentosa</td>
<td align="right">19</td>
<td align="right">334</td>
<td align="right">20</td>
<td align="right">90</td>
<td align="right">4.500</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">intra</td>
<td align="left">sarmentosa</td>
<td align="right">2</td>
<td align="right">103</td>
<td align="right">2</td>
<td align="right">20</td>
<td align="right">10.000</td>
</tr>
<tr class="odd">
<td align="left">sarmentosa</td>
<td align="left">mixed</td>
<td align="left">lydgatei</td>
<td align="right">5</td>
<td align="right"></td>
<td align="right">12</td>
<td align="right">63</td>
<td align="right">5.250</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">inter</td>
<td align="left">lydgatei</td>
<td align="right">7</td>
<td align="right">74</td>
<td align="right">9</td>
<td align="right">41</td>
<td align="right">4.556</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">intra</td>
<td align="left">stellarioides</td>
<td align="right">5</td>
<td align="right">192</td>
<td align="right">6</td>
<td align="right">26</td>
<td align="right">4.333</td>
</tr>
<tr class="even">
<td align="left">stellarioides</td>
<td align="left">mixed</td>
<td align="left">spergulina</td>
<td align="right">6</td>
<td align="right">137</td>
<td align="right">6</td>
<td align="right">27</td>
<td align="right">4.500</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">inter</td>
<td align="left">spergulina</td>
<td align="right">10</td>
<td align="right">163</td>
<td align="right">7</td>
<td align="right">13</td>
<td align="right">1.857</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">intra</td>
<td align="left">spergulina</td>
<td align="right">5</td>
<td align="right">159</td>
<td align="right">9</td>
<td align="right">67</td>
<td align="right">7.444</td>
</tr>
<tr class="odd">
<td align="left">spergulina</td>
<td align="left">mixed</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right"></td>
<td align="right">8</td>
<td align="right">50</td>
<td align="right">6.250</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">inter</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right">89</td>
<td align="right">3</td>
<td align="right">14</td>
<td align="right">4.667</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">intra</td>
<td align="left">membranacea</td>
<td align="right">4</td>
<td align="right">57</td>
<td align="right">5</td>
<td align="right">27</td>
<td align="right">5.400</td>
</tr>
<tr class="even">
<td align="left">membranacea</td>
<td align="left">mixed</td>
<td align="left">kauaiensis</td>
<td align="right">5</td>
<td align="right">206</td>
<td align="right">6</td>
<td align="right">60</td>
<td align="right">10.000</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">inter</td>
<td align="left">kauaiensis</td>
<td align="right">1</td>
<td align="right">153</td>
<td align="right">4</td>
<td align="right">18</td>
<td align="right">4.500</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">intra</td>
<td align="left">kauaiensis</td>
<td align="right">2</td>
<td align="right">1500</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">2.111</td>
</tr>
<tr class="odd">
<td align="left">kauaiensis</td>
<td align="left">mixed</td>
<td align="left">membranacea</td>
<td align="right">7</td>
<td align="right">240</td>
<td align="right">8</td>
<td align="right">63</td>
<td align="right">7.875</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">inter</td>
<td align="left">membranacea</td>
<td align="right">1</td>
<td align="right">242</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">3.000</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">intra</td>
<td align="left">kaalae</td>
<td align="right">7</td>
<td align="right">34</td>
<td align="right">20</td>
<td align="right">35</td>
<td align="right">1.750</td>
</tr>
<tr class="even">
<td align="left">kaalae</td>
<td align="left">mixed</td>
<td align="left">hookeri</td>
<td align="right">10</td>
<td align="right">93</td>
<td align="right">6</td>
<td align="right">34</td>
<td align="right">5.667</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">inter</td>
<td align="left">hookeri</td>
<td align="right">11</td>
<td align="right">175</td>
<td align="right">73</td>
<td align="right">118</td>
<td align="right">1.616</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">intra</td>
<td align="left">hookeri</td>
<td align="right">12</td>
<td align="right">35</td>
<td align="right">21</td>
<td align="right">128</td>
<td align="right">6.095</td>
</tr>
<tr class="odd">
<td align="left">hookeri</td>
<td align="left">mixed</td>
<td align="left">kaalae</td>
<td align="right">11</td>
<td align="right">189</td>
<td align="right">8</td>
<td align="right">70</td>
<td align="right">8.750</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">inter</td>
<td align="left">kaalae</td>
<td align="right">34</td>
<td align="right">156</td>
<td align="right">61</td>
<td align="right">107</td>
<td align="right">1.754</td>
</tr>
</tbody>
</table>
<pre class="r"><code>inventory %&gt;%  group_by(type) %&gt;% summarize(across(starts_with(&quot;crosses&quot;), sum, na.rm=T)) %&gt;% 
kable(caption=&quot;Number of crosses between individual plants used for seeds, germination and survival,
        and how many pots the seeds filled&quot;)</code></pre>
<table>
<caption>Number of crosses between individual plants used for seeds,
germination and survival, and how many pots the seeds filled</caption>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="right">crosses_surv</th>
<th align="right">crosses_seeds</th>
<th align="right">crosses_germ</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">intra</td>
<td align="right">60</td>
<td align="right">2337</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="right">78</td>
<td align="right">1230</td>
<td align="right">93</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="right">119</td>
<td align="right">1812</td>
<td align="right">208</td>
</tr>
</tbody>
</table>
<pre class="r"><code>inventory.momid &lt;- surv %&gt;% count(momsp, type, dadsp, momid) %&gt;% count(momsp, type, dadsp, name=&quot;crosses_surv&quot;) %&gt;% 
  full_join(germ %&gt;% count(momsp, type, dadsp, momid) %&gt;% count(momsp, type, dadsp, name=&quot;crosses_germ&quot;))
kable(inventory.momid, caption=&quot;Number of maternal plants used for seeds, germination and survival&quot;)</code></pre>
<table>
<caption>Number of maternal plants used for seeds, germination and
survival</caption>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="left">type</th>
<th align="left">dadsp</th>
<th align="right">crosses_surv</th>
<th align="right">crosses_germ</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">intra</td>
<td align="left">menziesii</td>
<td align="right">4</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">menziesii</td>
<td align="left">mixed</td>
<td align="left">salicaria</td>
<td align="right">7</td>
<td align="right">7</td>
</tr>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">inter</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">intra</td>
<td align="left">salicaria</td>
<td align="right">6</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">salicaria</td>
<td align="left">mixed</td>
<td align="left">menziesii</td>
<td align="right">12</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">inter</td>
<td align="left">menziesii</td>
<td align="right">10</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">intra</td>
<td align="left">lydgatei</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">lydgatei</td>
<td align="left">mixed</td>
<td align="left">sarmentosa</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">inter</td>
<td align="left">sarmentosa</td>
<td align="right">9</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">intra</td>
<td align="left">sarmentosa</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">sarmentosa</td>
<td align="left">mixed</td>
<td align="left">lydgatei</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">inter</td>
<td align="left">lydgatei</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">intra</td>
<td align="left">stellarioides</td>
<td align="right">5</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">stellarioides</td>
<td align="left">mixed</td>
<td align="left">spergulina</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">inter</td>
<td align="left">spergulina</td>
<td align="right">5</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">intra</td>
<td align="left">spergulina</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">spergulina</td>
<td align="left">mixed</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">inter</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">intra</td>
<td align="left">membranacea</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">membranacea</td>
<td align="left">mixed</td>
<td align="left">kauaiensis</td>
<td align="right">5</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">inter</td>
<td align="left">kauaiensis</td>
<td align="right">1</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">intra</td>
<td align="left">kauaiensis</td>
<td align="right">2</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">kauaiensis</td>
<td align="left">mixed</td>
<td align="left">membranacea</td>
<td align="right">7</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">inter</td>
<td align="left">membranacea</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">intra</td>
<td align="left">kaalae</td>
<td align="right">5</td>
<td align="right">12</td>
</tr>
<tr class="even">
<td align="left">kaalae</td>
<td align="left">mixed</td>
<td align="left">hookeri</td>
<td align="right">8</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">inter</td>
<td align="left">hookeri</td>
<td align="right">6</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">intra</td>
<td align="left">hookeri</td>
<td align="right">6</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">hookeri</td>
<td align="left">mixed</td>
<td align="left">kaalae</td>
<td align="right">9</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">inter</td>
<td align="left">kaalae</td>
<td align="right">9</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
<pre class="r"><code>inventory.momid %&gt;%  group_by(type) %&gt;% summarize(across(starts_with(&quot;crosses&quot;), sum, na.rm=T)) %&gt;% 
   kable(caption=&quot;Number of maternal plants used for seeds, germination and survival&quot;)</code></pre>
<table>
<caption>Number of maternal plants used for seeds, germination and
survival</caption>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="right">crosses_surv</th>
<th align="right">crosses_germ</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">intra</td>
<td align="right">44</td>
<td align="right">61</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="right">66</td>
<td align="right">61</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="right">60</td>
<td align="right">71</td>
</tr>
</tbody>
</table>
<pre class="r"><code># to run this, comment out across(matches(&quot;(mom|dad)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)) lines above
# left_join(count(germ, momsp, germid=momid, name=&quot;germ&quot;) %&gt;% mutate(momid=str_remove_all(germid, &quot;[ -]&quot;)),
#           germ %&gt;% group_by(momsp, germid=momid) %&gt;% summarize(tot.germ=sum(germinated))) %&gt;%
#   full_join(count(surv, momsp, survid=momid, name=&quot;surv&quot;)%&gt;% mutate(momid=str_remove_all(survid, &quot;[ -]&quot;))) %&gt;%
#   arrange(momsp, momid) %&gt;% select(momsp, momid, germid, survid, tot.germ, germ, surv) %&gt;%
#   add_count(momid) 
# #  write_tsv(&quot;data/survgerm_momids.tsv&quot;, na=&quot;&quot;)

mating &lt;- read_tsv(&quot;data/Pollen competition - mating.tsv&quot;) %&gt;% 
  left_join(tibble(momsp=sp_pairs, pair = factor(rep(1:5, each=2)))) %&gt;% 
  mutate(momsp = factor(momsp, level=sp_pairs)) 

styles &lt;- read_tsv(&quot;data/Pollen competition - style_length.tsv&quot;) %&gt;% 
  mutate(momsp = factor(sp_pairs[momsp], levels=sp_pairs))

seedsize &lt;- read_csv(&quot;data/seedlengths.csv&quot;) %&gt;% #from Schiedea monograph, SeedsNectaries project
  mutate(seedlength_mm = (seedlength_low_mm + seedlength_high_mm)/2) %&gt;%  #midpoint of range 
  rename(momsp=species) %&gt;% left_join(mating) %&gt;% drop_na(pair) %&gt;% group_by(pair) %&gt;% 
  mutate(momsp=factor(momsp, levels=sp_pairs), 
         seed_diff_mm = seedlength_mm-min(seedlength_mm)) %&gt;% arrange(momsp)</code></pre>
</div>
<div id="scoring-dates" class="section level1">
<h1>Scoring dates</h1>
<p>The date scored is not always available or accurate, do not use for
analysis.</p>
<pre class="r"><code>germ.pot %&gt;%  ggplot(aes(x=date_planted, y=date_scored, color=momsp)) + 
  geom_abline(slope=1, intercept=0) + geom_abline(slope=1, intercept=90) + geom_point() + geom_rug() +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(title=&quot;Germination scoring&quot;) #TODO check mis-entered dates</code></pre>
<p><img src="figures-comp/dates-1.svg" width="672" /></p>
<pre class="r"><code>germ.pot %&gt;% filter(abs(date_scored-date_planted)&lt;=90) %&gt;% ggplot(aes(x=date_scored - date_planted, fill=type)) +
  facet_wrap(vars(momsp)) +
  geom_histogram(binwidth=1) + scale_x_continuous(limits=c(0,NA)) + 
  labs(title=&quot;Germination scoring&quot;, subtitle = &quot;plot restricted to 90 days&quot;)</code></pre>
<p><img src="figures-comp/dates-2.svg" width="672" /></p>
<pre class="r"><code>surv %&gt;% ggplot(aes(x=date_planted, y=date_scored, color=momsp)) + geom_point() + geom_rug() + 
   scale_color_brewer(palette=&quot;Paired&quot;) + labs(title=&quot;Survival scoring&quot;)</code></pre>
<p><img src="figures-comp/dates-3.svg" width="672" /></p>
</div>
<div id="histograms" class="section level1">
<h1>Histograms</h1>
<pre class="r"><code>ggplot(seeds, aes(x=sds, fill=type)) + geom_histogram(binwidth=1) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-1.svg" width="672" /></p>
<pre class="r"><code>ggplot(germ, aes(x=planted, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-2.svg" width="672" /></p>
<pre class="r"><code>ggplot(germ, aes(x=germinated, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-3.svg" width="672" /></p>
<pre class="r"><code>ggplot(filter(surv, type !=&quot;inter&quot;), aes(x=sameasmom, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-4.svg" width="672" /></p>
<pre class="r"><code>ggplot(filter(surv, type !=&quot;intra&quot;), aes(x=hybrid, fill=type)) + geom_histogram(binwidth=1) + facet_wrap(vars(momsp))#lots of 20s</code></pre>
<p><img src="figures-comp/histogram-5.svg" width="672" /></p>
<pre class="r"><code>ggplot(surv, aes(x=dead, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp), scales=&quot;free_y&quot;)</code></pre>
<p><img src="figures-comp/histogram-6.svg" width="672" /></p>
<pre class="r"><code>pct_y &lt;- scale_y_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0,0.02))
pct_x &lt;- scale_x_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0,0))
germ.surv &lt;- full_join(germ, surv) #TODO fix matching

germ.surv %&gt;% mutate(germalive = paste(is.na(germinated), is.na(alive)) %&gt;% 
                       recode(`FALSE FALSE`=&quot;both&quot;, `FALSE TRUE`=&quot;survival&quot;, `TRUE FALSE`=&quot;germination&quot;)) %&gt;% 
  count(type, germalive) %&gt;% pivot_wider(names_from=&quot;germalive&quot;, values_from=&quot;n&quot;) %&gt;% 
  kable(kable=&quot;Matching crosses between germination and survival datasets&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="right">both</th>
<th align="right">germination</th>
<th align="right">survival</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">intra</td>
<td align="right">52</td>
<td align="right">8</td>
<td align="right">44</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="right">33</td>
<td align="right">45</td>
<td align="right">60</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="right">93</td>
<td align="right">26</td>
<td align="right">115</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggplot(germ.surv, aes(x=prop.germ, y=prop.alive, color=momsp, linetype=type)) + 
  geom_point() + geom_smooth(se=F, span=1.1) + pct_y + pct_x + theme(legend.position=&quot;top&quot;) + 
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Germination&quot;,y=&quot;Survival&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-1.svg" width="672" /></p>
<pre class="r"><code>ggplot(germ.surv %&gt;% filter(type==&quot;mixed&quot;), aes(x=prop.alive, y=prop.hybrid, color=momsp)) + 
  geom_point() + geom_smooth(se=F, span=1.1)+ pct_y + pct_x + theme(legend.position=&quot;top&quot;) +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Survival&quot;,y=&quot;Hybrids&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-2.svg" width="672" /></p>
<pre class="r"><code>ggplot(germ.surv %&gt;% filter(type==&quot;mixed&quot;), aes(x=prop.germ, y=prop.hybrid, color=momsp)) + 
  geom_point() + geom_smooth(se=F, span=1.1)+pct_y + pct_x + theme(legend.position=&quot;top&quot;) +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Germination&quot;,y=&quot;Hybrids&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-3.svg" width="672" /></p>
<pre class="r"><code>#does germ and surv not being independent make a difference when you take the mean of each and then multiply, versus
# multiplying within each cross?
# germ.surv %&gt;%   mutate(prop.germsurv = prop.germ * prop.alive) %&gt;% drop_na(prop.germsurv) %&gt;% #multiply within each cross
#   group_by(momsp) %&gt;% summarize(across(starts_with(&quot;prop&quot;), mean)) %&gt;% 
#   mutate(prop.germXsurv = prop.germ * prop.alive) %&gt;% #multiply after taking the mean
#   ggplot(aes(x=prop.germsurv, y= prop.germXsurv, color=momsp)) + scale_color_brewer(palette=&quot;Paired&quot;)+
#   geom_point() + geom_abline(intercept=0, slope=1)
# 
# #try it with perfectly correlated data
# fakecrosses &lt;- tibble(cross=0:10, germ = cross/10, surv=germ, germsurv=germ * surv)
# fakecrosses %&gt;% summarize(across(everything(), mean)) %&gt;% mutate(germXsurv = germ * surv)</code></pre>
</div>
<div id="distribution-of-cumulative-fitness" class="section level1">
<h1>Distribution of cumulative fitness</h1>
<pre class="r"><code>#dataname = name of the dataset, yvar = the fitness trait, N = number of trials for each species and crosstype
sample_trait &lt;- function(dataname, yvar, N) {
  get(dataname) %&gt;% select(sxc, y=!!yvar) %&gt;% 
    drop_na(y) %&gt;% #drop missing trait values (various reasons for these)
    group_by(sxc) %&gt;% nest() %&gt;% #subset data by the species and crosstype (between/within/hybrid)
    mutate(sampled = map(data, slice_sample, n=N, replace=T)) %&gt;% #sample each subset N times with replacement
    select(-data) %&gt;% unnest(sampled) %&gt;% arrange(sxc) %&gt;% mutate(trial=row_number()) #give each trial an index
}
traits &lt;- enframe(propnames[2:4], name=&quot;yvar&quot;, value=&quot;ylab&quot;) %&gt;%
  mutate(data = c(&quot;seeds&quot;,&quot;germ&quot;,&quot;surv&quot;)) #seeds includes zeros (no capsule formed)

set.seed(1)
ntrials &lt;- 1000
F1.cumul.fitness &lt;- traits %&gt;% select(data, yvar) %&gt;%  # independent multiplicative traits
  mutate(samp = pmap(., sample_trait, N = ntrials)) %&gt;% # sample each trait N times per species and crosstype
  select(-data) %&gt;% unnest(samp) %&gt;% # expand sampled dataset to full size
  pivot_wider(id_cols=c(sxc, trial), names_from=yvar, values_from=y) %&gt;% # line up the trials for each variable
  drop_na(sds, prop.germ, prop.alive) %&gt;% #drop missing data for sarm+sper mixed pollinations
  mutate(fitness_sds         = sds,
         fitness_prop.germ   = sds * prop.germ,
         fitness_prop.alive  = sds * prop.germ * prop.alive)

F1.cumul.fitness %&gt;% select(sxc, starts_with(&quot;fitness&quot;)) %&gt;% 
  pivot_longer(starts_with(&quot;fitness&quot;), names_to=&quot;stage&quot;, names_prefix = &quot;fitness_&quot;) %&gt;% 
  separate(sxc, into=c(&quot;momsp&quot;,&quot;type&quot;)) %&gt;% 
  mutate(type = factor(type, levels=names(typenames)),
         stage = factor(stage, levels=traits$yvar), momsp=factor(momsp, levels=sp_pairs)) %&gt;% 
  ggplot(aes(x=value, fill=stage)) + 
  facet_grid(type~momsp, scales=&quot;free_x&quot;, labeller=as_labeller(c(set_names(sp_pairs), typenames)))+
  geom_histogram(binwidth=1, boundary=-0.5, position=position_identity(), linewidth=1, alpha=0.5) +
  scale_y_continuous(labels = ~scales::percent(./ntrials)) + 
  scale_fill_hue(&quot;Cumulative fitness after this stage: &quot;,labels=traits$ylab) +
  labs(y=paste0(&quot;Frequency of outcome (&quot;,ntrials,&quot; trials)&quot;), 
       x=&quot;Surviving hybrids resulting from one pollination&quot;) +
  theme_minimal() + theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figures-comp/totalRI_simulation-1.svg" width="768" /></p>
<pre class="r"><code>F1.cumul.fitness %&gt;% group_by(sxc) %&gt;% summarize(prop.zero=mean(fitness_sds&gt;0)) %&gt;% 
ggplot(aes(x=prop.zero, y=sxc)) + 
  geom_point(size=2)+ 
  scale_x_continuous(limits=c(0,1), labels=scales::percent) + theme_minimal() + 
  labs(x=paste0(&quot;Nonzero F1 flowers resulting from one P x P pollination (&quot;,ntrials,&quot; trials)&quot;), y=&quot;Crosstype&quot;,
       title = &quot;Frequency cumulativefitness greater than zero&quot;) </code></pre>
<p><img src="figures-comp/totalRI_simulation-2.svg" width="768" /></p>
<pre class="r"><code>trials_to_average &lt;- 50
calc_RI &lt;- function(h, b) 1 - 2 *  h/(h + b) #method of Sobel and Chen 2014 for hybrids vs. between-population intraspecific fitness, varies between -1 (hybrd vigor) and 1 (complete isolation)
F1.cumul.isolation &lt;- F1.cumul.fitness %&gt;% 
  group_by(sxc, meanblock = ceiling(trial / trials_to_average)) %&gt;% #group into blocks of trials_to_average trials
  summarize(across(starts_with(&quot;fitness&quot;), mean)) %&gt;% #take the mean fitness for each block and crosstype
  pivot_longer(starts_with(&quot;fitness&quot;), names_prefix = &quot;fitness_&quot;, names_to=&quot;stage&quot;, values_to=&quot;fitness&quot;) %&gt;% 
  separate(sxc, into=c(&quot;momsp&quot;,&quot;crosstype&quot;)) %&gt;% 
  pivot_wider(names_from=crosstype, values_from=fitness) %&gt;% 
  mutate(stage = factor(stage, levels=unique(traits$yvar)),
         RI = calc_RI(h=intra, b=inter)) 

ggplot(F1.cumul.isolation, aes(y=fct_rev(stage), color=factor(momsp, levels=sp_pairs), x=RI)) + 
  geom_vline(xintercept=0, linetype=2) + geom_vline(xintercept=-1)+ geom_vline(xintercept=1)+ 
  geom_boxplot(position=position_dodge(width=0.5), outlier.size = 0.2) + 
  scale_y_discrete(labels=deframe(select(traits, yvar, ylab))) +
  scale_color_brewer(palette = &quot;Paired&quot;) +
  scale_x_continuous(limits=c(-1.1,1.1), breaks=c(-1,-0.5,0,0.5,1),
                     labels=c(&quot;-1\nheterosis&quot;,&quot;-0.5&quot;,&quot;0\nno\nbarrier&quot;,&quot;0.5&quot;,&quot;1\ncomplete\nisolation&quot;)) + 
  theme_minimal() + theme(legend.position = &quot;top&quot;, legend.text = element_text(face=&quot;italic&quot;)) +
  labs(x=&quot;Cumulative reproductive isolation&quot;, y=&quot;Fitness component&quot;, color=&quot;Maternal parent&quot;)</code></pre>
<p><img src="figures-comp/totalRI_means-1.svg" width="624" /></p>
</div>
<div id="distribution-of-estimated-hybrid-proportions"
class="section level1">
<h1>Distribution of estimated hybrid proportions</h1>
<pre class="r"><code>#make a version of the resampled data that avoids zero proportions
set.seed(1)
ntrials &lt;- 1000
F1.cumul.fitness.nz &lt;- traits %&gt;% select(data, yvar) %&gt;%  # independent multiplicative traits
  mutate(yvar = paste0(yvar, &quot;.nz&quot;)) %&gt;% #get the data without zeros (add 1 to num and denom)
  mutate(samp = pmap(., sample_trait, N = ntrials)) %&gt;% # sample each trait N times per species and crosstype
  select(-data) %&gt;% unnest(samp) %&gt;% # expand sampled dataset to full size
  pivot_wider(id_cols=c(sxc, trial), names_from=yvar, values_from=y) %&gt;% # line up the trials for each variable
  drop_na(sds.nz, prop.germ.nz, prop.alive.nz) #drop missing data for sarm+sper mixed pollinations

#get Monte Carlo samples for uncertainty estimation
samples_per_batch &lt;- 20
F1.backest &lt;- surv %&gt;% filter(type==&quot;mixed&quot;) %&gt;% select(momsp, hybrid, sameasmom) %&gt;% mutate(batch = row_number()) %&gt;%
  pivot_longer(c(hybrid, sameasmom), names_to=&quot;type&quot;, values_to = &quot;plt&quot;) %&gt;% 
  mutate(type = recode(type, hybrid=&quot;inter&quot;, sameasmom=&quot;intra&quot;)) %&gt;% 
  left_join(F1.cumul.fitness.nz %&gt;% select(sxc, prop.alive.nz, prop.germ.nz, sds.nz) %&gt;% 
              separate(sxc, into=c(&quot;momsp&quot;,&quot;type&quot;)) %&gt;% group_by(momsp, type) %&gt;% nest(), by=c(&quot;momsp&quot;, &quot;type&quot;)) %&gt;% 
  mutate(data = map(data, ~mutate(sample_n(.x, samples_per_batch), samp=row_number()))) %&gt;% unnest(data) %&gt;% 
  mutate(sdl = plt/prop.alive.nz, #back-estimate but avoid dividing by zero by adding one to num and den
         sds = sdl/prop.germ.nz, 
         zyg = sds/sds.nz) %&gt;% #add one to all seed counts to avoid zeros
  pivot_wider(names_from=&quot;type&quot;, values_from=c(plt, prop.alive.nz, prop.germ.nz, sds.nz, sdl, sds, zyg)) %&gt;% 
  mutate(hyb.plt = plt_inter / (plt_inter + plt_intra),
         hyb.sdl = sdl_inter / (sdl_inter + sdl_intra),
         hyb.sds = sds_inter / (sds_inter + sds_intra),
         hyb.zyg = zyg_inter / (zyg_inter + zyg_intra))

F1.backest.long &lt;- F1.backest %&gt;% 
  select(momsp, batch, samp, starts_with(&quot;hyb&quot;)) %&gt;% 
  mutate(momsp= factor(momsp, levels=sp_pairs)) %&gt;% 
  pivot_longer(starts_with(&quot;hyb&quot;), names_prefix=&quot;hyb.&quot;, names_to=&quot;stage&quot;, values_to=&quot;prop&quot;) %&gt;% 
  arrange(momsp, batch, stage)

ggplot(F1.backest.long, aes(x=fct_rev(stage), color=momsp, y=prop)) + 
  facet_wrap(vars(momsp), ncol=2) +
  geom_path(aes(group=paste(batch, samp)), alpha=0.2)+ 
  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) + geom_jitter(size=0.2, height=0, width=0.2)  +
  scale_color_brewer(palette = &quot;Paired&quot;) + guides(color=&quot;none&quot;) +
  scale_x_discrete(labels=c(zyg=&quot;Comp&quot;, sds=&quot;Seed&quot;, sdl=&quot;Seedling&quot;, plt=&quot;Plant&quot;))+
  labs(x=&quot;Life stage&quot;, y=&quot;Proportion of hybrids&quot;)</code></pre>
<p><img src="figures-comp/backwards-1.svg" width="768" /></p>
</div>
<div id="models" class="section level1">
<h1>Models</h1>
<p>Expand this code to see the model specifications. The models have the
following settings:</p>
<ul>
<li>response variable: proportion seeds that germinated, proportion
seedlings that survived, proportion of live seedlings that are
hybrids</li>
<li>sxc: the combined maternal plant species and crosstype (within
species, between species, mixed pollen)</li>
<li>momid: random effects of the maternal plant genotype</li>
<li>family: betabinomial (for binary variables)</li>
</ul>
<pre class="r"><code>comp.mod &lt;- list(
  capsule.formed=glmmTMB(capsule.formed ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=seeds %&gt;% mutate(sxc=factor(sxc))),
  sds =          glmmTMB(sds            ~ sxc + (1|momid), family=&quot;truncated_nbinom1&quot;, data=filter(seeds, sds&gt;0) %&gt;% mutate(sxc=factor(sxc))),
  prop.germ =    glmmTMB(prop.germ   ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=germ %&gt;% mutate(sxc=factor(sxc)), weights = planted),
  prop.alive =   glmmTMB(prop.alive  ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=surv %&gt;% mutate(sxc=factor(sxc)), weights = total),
  prop.hybrid =  glmmTMB(prop.hybrid ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=filter(surv, type==&quot;mixed&quot;) %&gt;% mutate(sxc=factor(sxc)), weights = alive))</code></pre>
<div id="maternal-effects" class="section level2">
<h2>Maternal effects</h2>
<pre class="r"><code>comp.ranef &lt;- map_dfr(comp.mod, ~as.data.frame(ranef(.x)), .id=&quot;trait&quot;) %&gt;% as_tibble() %&gt;% 
  rename(momid=grp, estimate = condval, sd = condsd) %&gt;% select(-component, -grpvar, -term) %&gt;% 
  mutate(momid= as.character(momid), trait=factor(trait, levels=names(propnames))) %&gt;% 
  left_join(count(germ, momsp.germ=momsp, momid, name=&quot;germ&quot;)) %&gt;% 
  left_join(count(surv, momsp.surv=momsp, momid, name=&quot;surv&quot;)) %&gt;% 
  left_join(count(seeds, momsp.seeds=momsp, momid, name=&quot;seeds&quot;)) %&gt;%
  mutate(momsp = coalesce(coalesce(momsp.germ, momsp.surv),momsp.seeds), .keep=&quot;unused&quot;)

ggplot(comp.ranef, aes(x=estimate, y=paste(str_sub(momsp,1,4),momid), color=momsp)) + 
  facet_wrap(vars(trait), labeller = as_labeller(propnames), nrow=1) +
  geom_pointrange(aes(xmin=estimate-sd, xmax=estimate+sd))  + theme(axis.text.y=element_text(size=7))+
  scale_color_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) +  labs(x=&quot;Maternal effect&quot;, y=&quot;Maternal plant&quot;) </code></pre>
<p><img src="figures-comp/ranef-1.svg" width="672" /></p>
<pre class="r"><code>comp.ranef %&gt;% select(-sd) %&gt;% pivot_wider(names_from=&quot;trait&quot;, values_from=&quot;estimate&quot;) %&gt;%
  ggplot(aes(y=prop.alive, x=prop.germ, color=momsp)) + geom_point() + scale_color_brewer(palette=&quot;Paired&quot;) +
  geom_smooth(aes(group=1), color=&quot;grey30&quot;) + theme(legend.position=&quot;top&quot;)+
  labs(x=&quot;Maternal effect on germination&quot;, y=&quot;Maternal effect on survival&quot;, color=&quot;Maternal species&quot;)</code></pre>
<p><img src="figures-comp/ranef-2.svg" width="672" /></p>
<pre class="r"><code>comp.ranef %&gt;% select(-sd) %&gt;% pivot_wider(names_from=&quot;trait&quot;, values_from=&quot;estimate&quot;) %&gt;%
  ggplot(aes(y=prop.hybrid, x=prop.alive, color=momsp)) + geom_point() + scale_color_brewer(palette=&quot;Paired&quot;) +
  geom_smooth(aes(group=1), color=&quot;grey30&quot;) + theme(legend.position=&quot;top&quot;)+
  labs(x=&quot;Maternal effect on survival&quot;, y=&quot;Maternal effect on hybrids&quot;, color=&quot;Maternal species&quot;)</code></pre>
<p><img src="figures-comp/ranef-3.svg" width="672" /></p>
</div>
<div id="inference" class="section level2">
<h2>Inference</h2>
<pre class="r"><code>comp.tests &lt;- map_dfr(comp.mod[-5], ~multcomp::glht(.x, linfct = multcomp::mcp(sxc = paste0(sp_pairs,&quot;.inter - &quot;,sp_pairs,&quot;.intra = 0&quot;))) %&gt;% 
                          multcomp:::summary.glht(test=multcomp::adjusted(type=&quot;fdr&quot;)) %&gt;% tidy(), .id=&quot;trait&quot;) %&gt;% 
    mutate(momsp = factor(str_extract(contrast, &quot;([^\\.]+)&quot;), levels=sp_pairs), 
           trait = factor(trait, levels=names(propnames))) %&gt;% 
  select(trait, momsp, inter.minus.intra=estimate, std.error, adj.p.value)
  
comp.emm &lt;-  map_dfr(comp.mod, ~emmeans(., ~sxc) %&gt;% summary(type=&quot;response&quot;) %&gt;%  #average over collect date
                             tidy %&gt;% rename(estimate=any_of(c(&quot;response&quot;,&quot;prob&quot;))), .id=&quot;trait&quot;) %&gt;% 
  separate(sxc, into = c(&quot;momsp&quot;,&quot;type&quot;)) %&gt;% 
  mutate(momsp=factor(momsp, levels=sp_pairs), 
         type=factor(type, levels= names(typenames)),
         trait=factor(trait, levels=names(propnames)))

calc_RI &lt;- function(h, b) 1 - 2 *  h/(h + b) #method of Sobel and Chen 2014 for hybrids vs. between-population intraspecific fitness, varies between -1 (hybrd vigor) and 1 (complete isolation)
highlow &lt;- function(fn, a, b) { #method of Sobel and Streisfeld 2015 Fig. S1
  combos &lt;- c(fn(a[1], b[1]), fn(a[1],b[2]), fn(a[2],b[1]), fn(a[2], b[2]))
  set_names(c(min(combos), max(combos)), c(&quot;low&quot;,&quot;high&quot;))
}
postzygotic.ratios &lt;- comp.emm %&gt;% filter(type %in% c(&quot;inter&quot;,&quot;intra&quot;)) %&gt;% 
  select(trait, momsp, type, estimate, conf.low, conf.high) %&gt;% 
  pivot_wider(names_from = type, values_from = c(estimate, conf.low, conf.high)) %&gt;% 
  rename_with(~ str_remove(.,&quot;estimate_|conf.&quot;)) %&gt;% rowwise() %&gt;% 
  mutate(across(c(intra, inter), ~ ./max(intra, inter)),
         hybrid.vigor = inter/intra,
         RI = calc_RI(h=inter, b=intra)) %&gt;% 
  mutate(RI.range = list(highlow(fn=calc_RI, a=c(low_inter, high_inter), b=c(low_intra, high_intra))), 
         .keep=&quot;unused&quot;) %&gt;% unnest_wider(RI.range)

ggplot(postzygotic.ratios, aes(y=fct_rev(momsp), x=trait, fill=log2(hybrid.vigor))) +
  geom_tile() + geom_text(aes(label=format(hybrid.vigor,digits=1))) + 
  scale_fill_gradient2(guide=&quot;none&quot;) + scale_x_discrete(labels=propnames, position=&quot;top&quot;)+
  theme(axis.title = element_blank(), axis.text.y=element_text(color=rev(sp_pal_dark), face=&quot;italic&quot;))</code></pre>
<p><img src="figures-comp/inference-1.svg" width="672" /></p>
<pre class="r"><code>#doesn&#39;t work, compares all to the first level (hookeri), and glht doesn&#39;t support comparison to a constant
#hybrid.half &lt;- multcomp::glht(comp.mod$prop.hybrid, multcomp::mcp(sxc = paste0(sp_pairs[-7],&quot;.mixed = 0&quot;)))
#compare to null of 50% hybrids, since pollen mixtures were 50-50

# function that takes p = proportion of hybrids and 
# W = relative fitness of hybrids = absolute fitness of hybrids / absolute fitness of intraspecifics
# and returns the proportion of hybrids in the previous life stage.
# based on solving the following for pp (previous proportion):
# p = pp W / (pp W + 1 - pp)
# 1 - pp represents the previous proportion of intraspecifics
# see https://www.zoology.ubc.ca/~otto/PopGen500/Lecture1/Overheads.html
goback &lt;- function(p, W) p/(p + W - p*W)

hybrid.test &lt;- surv %&gt;% filter(type==&quot;mixed&quot;) %&gt;% 
  group_by(momsp, type, dadsp) %&gt;% 
  summarize(total=sum(total), alive=sum(alive), hybrid=sum(hybrid), crosses=n(), .groups=&quot;drop&quot;) %&gt;% 
  mutate(total_per_c = total/crosses, alive_per_c = alive / crosses,
         prop.hybrid = hybrid/alive,
         p = map2_dbl(hybrid, alive, ~binom.test(x=.x, n=.y, p = 0.5)$p.value),
         p.adj = p.adjust(p, &quot;bonferroni&quot;),
         stars = cut(p.adj, breaks = c(-Inf, 0.01, 0.05, 0.10, Inf), 
                labels = c(&quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;n.s.&quot;), right = FALSE)) %&gt;% 
  left_join(postzygotic.ratios %&gt;% pivot_wider(id_cols = momsp, names_from=trait, values_from = hybrid.vigor)) %&gt;% 
  mutate(hybrid.vigor.all = capsule.formed * sds * prop.germ * prop.alive, #these are all inter/intra relative fitnesses (hybrid vigor)
         prop.hybrid.sdl = goback(prop.hybrid, prop.alive), #seedling stage
         prop.hybrid.sds = goback(prop.hybrid.sdl, prop.germ), #seed stage
         prop.hybrid.zyg = goback(goback(prop.hybrid.sds, sds), capsule.formed)) #two steps!

hybnames &lt;- c(prop.hybrid =     &quot;Plant&quot;, #TODO rename this hybnames to avoid conflict
               prop.hybrid.sdl = &quot;Seedling&quot;,
               prop.hybrid.sds = &quot;Seed&quot;,
               prop.hybrid.zyg = &quot;Competition&quot;)

hybrid.test %&gt;% select(momsp, all_of(names(hybnames))) %&gt;% 
  pivot_longer(-momsp, names_to=&quot;trait&quot;) %&gt;% 
  ggplot(aes(y=fct_rev(momsp), x=trait, fill=log2(value))) +
  geom_tile() + geom_text(aes(label=format(value,digits=1))) + scale_fill_gradient2(guide=&quot;none&quot;, midpoint=-1) + 
  scale_x_discrete(labels=hybnames, position=&quot;top&quot;)+
  theme(axis.title = element_blank(), axis.text.y=element_text(color=rev(sp_pal_dark), face=&quot;italic&quot;))</code></pre>
<p><img src="figures-comp/inference-2.svg" width="672" /></p>
<pre class="r"><code>hybrid.test %&gt;% select(momsp, all_of(names(hybnames))) %&gt;% 
  pivot_longer(-momsp, names_to=&quot;trait&quot;) %&gt;% mutate(trait = factor(trait, levels = rev(names(hybnames)))) %&gt;% 
  ggplot(aes(color=momsp, x=trait, y=value, group=momsp)) + 
  geom_hline(yintercept=0.5, linetype=2)+
  geom_path(linewidth=2) + geom_point(size=4) +
  scale_color_brewer(&quot;Maternal species&quot;, palette=&quot;Paired&quot;) +
  scale_x_discrete(&quot;Life stage&quot;, labels=hybnames, position=&quot;top&quot;)+
  scale_y_continuous(&quot;Estimated proportion of hybrid progeny&quot;, breaks=0:10/10)+
  theme(legend.text=element_text(face=&quot;italic&quot;), panel.grid.major.x = element_blank()) </code></pre>
<p><img src="figures-comp/inference-3.svg" width="672" /></p>
<pre class="r"><code>asym.test &lt;- surv %&gt;% filter(type==&quot;mixed&quot;) %&gt;% 
  left_join(mating) %&gt;% arrange(pair) %&gt;% group_by(pair) %&gt;% nest() %&gt;% 
  mutate(model = map(data, ~glmmTMB(prop.hybrid ~ momsp, family=&quot;betabinomial&quot;, data=., weights = alive)),
         anova = map(model, ~tidy(car::Anova(.x)))) %&gt;% #(1|momid) broke model for 3rd pair
  select(-data, -model) %&gt;% unnest(anova) %&gt;% filter(term !=&quot;Residuals&quot;) %&gt;% 
  left_join(summarize(mating, species=paste(momsp, collapse=&quot;-&quot;), .by=pair))
kable(asym.test, caption=&quot;Differences in hybrid proportions by cross direction&quot;)</code></pre>
<table>
<caption>Differences in hybrid proportions by cross direction</caption>
<thead>
<tr class="header">
<th align="left">pair</th>
<th align="left">term</th>
<th align="right">statistic</th>
<th align="right">df</th>
<th align="right">p.value</th>
<th align="left">species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">momsp</td>
<td align="right">5.1712</td>
<td align="right">1</td>
<td align="right">0.0230</td>
<td align="left">menziesii-salicaria</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">momsp</td>
<td align="right">23.6881</td>
<td align="right">1</td>
<td align="right">0.0000</td>
<td align="left">lydgatei-sarmentosa</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">momsp</td>
<td align="right">12.9453</td>
<td align="right">1</td>
<td align="right">0.0003</td>
<td align="left">stellarioides-spergulina</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">momsp</td>
<td align="right">0.0648</td>
<td align="right">1</td>
<td align="right">0.7991</td>
<td align="left">membranacea-kauaiensis</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">momsp</td>
<td align="right">4.1437</td>
<td align="right">1</td>
<td align="right">0.0418</td>
<td align="left">hookeri-kaalae</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="plots" class="section level1">
<h1>Plots</h1>
<p>Boxplots of the germination and survival proportions in each plant x
plant cross. Germination proportions are calculated across all pots of a
given plant x plant cross. Points show estimated marginal mean and 95%
CI.</p>
<pre class="r"><code>sp_theme &lt;- theme(legend.position = &quot;top&quot;, 
                  panel.grid.major.x=element_blank(), panel.grid.minor.y=element_blank(), 
                  axis.text.x = element_text(angle=20, face=&quot;italic&quot;), axis.text = element_text(color=&quot;black&quot;))
bxplt &lt;- geom_boxplot(position = position_dodge(width=0.9), color=&quot;grey50&quot;, outlier.size=0.5)
alternate &lt;- geom_tile(data=tibble(momsp=sp_pairs[seq(1,10,by=2)]), aes(x=momsp, y=0.5, height=1, width=1),  alpha=0.4, fill=&quot;grey&quot;, inherit.aes = F) 

ggplot(filter(seeds.bycross, !momsp %in% c(&quot;sarmentosa&quot;,&quot;spergulina&quot;)), aes(x=momsp, fill=type, y=capsule.formed))  +
  bxplt + alternate + bxplt +
  geom_pointrange(data=filter(comp.emm, trait==&quot;capsule.formed&quot;, !momsp %in% c(&quot;sarmentosa&quot;,&quot;spergulina&quot;)), aes(y=estimate, ymax=conf.high, ymin=conf.low, color=type),
                  position = position_dodge(width=0.9), size=0.5, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) + scale_color_manual(values=type_pal_dark, labels=typenames) + 
  scale_x_discrete(position=&quot;top&quot;, drop=FALSE) +  pct_y +
  labs(x=&quot;Maternal species&quot;, y=&quot;Capsule formation&quot;, fill=&quot;Pollination&quot;) + sp_theme </code></pre>
<p><img src="figures-comp/rawdata-1.svg" width="672" /></p>
<pre class="r"><code>ggplot(seeds %&gt;% filter(sds&gt;0), aes(x=momsp, fill=type, y=sds))  + bxplt + 
  geom_tile(data=tibble(momsp=sp_pairs[seq(1,10,by=2)]), aes(x=momsp, y=35, height=70, width=1), 
            alpha=0.2, fill=&quot;grey&quot;, inherit.aes = F) + bxplt +
  geom_pointrange(data=filter(comp.emm, trait==&quot;sds&quot;), aes(y=estimate, ymax=conf.high, ymin=conf.low, color=type),
                  position = position_dodge(width=0.9), size=0.5, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) + scale_color_manual(values=type_pal_dark, labels=typenames) +
  scale_y_continuous(expand=c(0.02,0))+
  labs(x=&quot;Maternal species&quot;, y=&quot;Seeds&quot;, fill=&quot;Pollination&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-2.svg" width="672" /></p>
<pre class="r"><code>ggplot(germ, aes(x=momsp, fill=type, y=prop.germ))   + bxplt + alternate + bxplt + 
  geom_pointrange(data=filter(comp.emm, trait==&quot;prop.germ&quot;), aes(y=estimate, ymax=conf.high, ymin=conf.low, color=type),
                  position = position_dodge(width=0.9), size=0.5, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) + scale_color_manual(values=type_pal_dark, labels=typenames) +
  scale_x_discrete(position=&quot;top&quot;) +  pct_y +
  labs(x=&quot;Maternal species&quot;, y=&quot;Germination&quot;, fill=&quot;Pollination&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-3.svg" width="672" /></p>
<pre class="r"><code>ggplot(surv, aes(x=momsp, fill=type, y=prop.alive)) + bxplt + alternate + bxplt +
  geom_pointrange(data=filter(comp.emm, trait==&quot;prop.alive&quot;), aes(y=estimate, ymax=conf.high, ymin=conf.low, color=type),
                  position = position_dodge(width=0.9), size=0.5, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) +  scale_color_manual(values=type_pal_dark, labels=typenames) +
  scale_x_discrete(position=&quot;top&quot;) +  pct_y +
  labs(x=&quot;Maternal species&quot;, y=&quot;Survival&quot;, fill=&quot;Pollination&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-4.svg" width="672" /></p>
<pre class="r"><code>ggplot(filter(surv, type==&quot;mixed&quot;), aes(x=momsp, y=prop.hybrid, fill=momsp)) + 
  geom_hline(yintercept=0.5)+ geom_violin(scale=&quot;width&quot;, draw_quantiles = c(0.25,0.5,0.75))+ 
  geom_pointrange(data=left_join(filter(comp.emm, trait==&quot;prop.hybrid&quot;), hybrid.test),
                  aes(y=estimate, ymax=conf.high, ymin=conf.low, shape=p.adj &lt; 0.05), size=1) + 
  scale_shape_manual(&quot;Different from 50%&quot;, values=c(1, 19), labels=c(&quot;no&quot;,&quot;yes&quot;))+
  geom_text(data=hybrid.test, aes(y=1.1, label=crosses))+ annotate(&quot;text&quot;, x=11, y=1.1, label=&quot;crosses&quot;)+
  geom_text(data=hybrid.test, aes(y=1.04, label=alive))+ annotate(&quot;text&quot;, x=11, y=1.04, label=&quot;progeny&quot;)+
  coord_cartesian(clip=&quot;off&quot;)+
  scale_fill_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) +
  scale_y_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0.02,0), limits=c(0,1.1))+
  labs(x=&quot;Maternal species&quot;, y=&quot;Hybrids from competitive pollinations&quot;) + sp_theme + 
  theme(plot.margin = unit(5.5*c(1,6,1,1), &quot;points&quot;))</code></pre>
<p><img src="figures-comp/rawdata-5.svg" width="672" /></p>
<pre class="r"><code>hybrid.test %&gt;% left_join(filter(comp.emm, trait==&quot;prop.hybrid&quot;)) %&gt;% 
  select(momsp, dadsp, prop.hybrid, std.error, p.adj) %&gt;% kable(caption=&quot;Test if hybrid proportions differ from 50%&quot;)</code></pre>
<table>
<caption>Test if hybrid proportions differ from 50%</caption>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="left">dadsp</th>
<th align="right">prop.hybrid</th>
<th align="right">std.error</th>
<th align="right">p.adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">salicaria</td>
<td align="right">0.4148</td>
<td align="right">0.0713</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">menziesii</td>
<td align="right">0.5457</td>
<td align="right">0.0617</td>
<td align="right">0.0274</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">sarmentosa</td>
<td align="right">0.1045</td>
<td align="right">0.0482</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">lydgatei</td>
<td align="right">0.5390</td>
<td align="right">0.1011</td>
<td align="right">1.0000</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">spergulina</td>
<td align="right">0.0386</td>
<td align="right">0.0309</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">stellarioides</td>
<td align="right">0.3860</td>
<td align="right">0.1000</td>
<td align="right">0.0001</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">kauaiensis</td>
<td align="right">0.0129</td>
<td align="right">0.0223</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">membranacea</td>
<td align="right">0.0331</td>
<td align="right">0.0171</td>
<td align="right">0.0000</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">hookeri</td>
<td align="right">0.1933</td>
<td align="right">0.0413</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">kaalae</td>
<td align="right">0.3021</td>
<td align="right">0.0621</td>
<td align="right">0.0000</td>
</tr>
</tbody>
</table>
<p>Tested whether the proportion of hybrids significantly differed from
50%, accounting for multiple comparisons. Violin plots show proportions
for each plant x plant cross. The number of plant x plant crosses
indicated above the plot.</p>
<div id="predicted-and-actual-germination" class="section level2">
<h2>Predicted and actual germination</h2>
<pre class="r"><code>#hybrid.vigor.germsurv = (prop.germ_inter / prop.germ_intra) * (prop.surv_inter / prop.surv_intra)
#prop.hybrid.sds = prop.hybrid / hybrid.vigor.germsurv
# prop.germ_mixed.pred ?= prop.germ_mixed
# prop.germ_mixed.pred = prop.hybrid.sds * prop.germ_inter + (1-prop.hybrid.sds) * prop.germ_intra
#                      = prop.germ_intra + prop.hybrid.sds * (prop.germ_inter - prop.germ_intra)

hybrid.test %&gt;% left_join(pivot_wider(comp.emm, id_cols = momsp, names_from=c(trait, type), values_from = estimate)) %&gt;% 
  mutate(prop.germ_mixed.pred = prop.hybrid.sds * prop.germ_inter + (1-prop.hybrid.sds) * prop.germ_intra) %&gt;% 
  ggplot(aes(x = momsp, color=momsp)) + 
  geom_segment(aes(y=prop.germ_intra, yend=prop.germ_inter, xend=momsp))+
  geom_point(aes(y=prop.germ_inter, shape=&quot;inter&quot;), size=4)+
  geom_point(aes(y=prop.germ_intra, shape=&quot;intra&quot;), size=4)+
  geom_point(aes(y=prop.germ_mixed, shape=&quot;mixed&quot;), size=4)+
  geom_point(aes(y=prop.germ_mixed.pred, shape=&quot;pred&quot;), size=4)+
  geom_label(aes(y=prop.germ_mixed.pred, label=paste0(round(prop.hybrid.sds*100),&quot;%&quot;), fill=momsp), show.legend = F, nudge_x = 0.5, color=&quot;black&quot;) +
  scale_shape_manual(&quot;Cross&quot;, values=c(inter=15, intra=16, mixed=17, pred=3), 
                     labels=c(typenames, pred=&quot;Predicted from estimated hybrid seed ratio&quot;))+
  scale_color_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) + scale_fill_brewer(palette=&quot;Paired&quot;) +
  scale_y_continuous(&quot;Germination&quot;, limits=c(0,1), expand=c(0,0), labels=~scales::percent(.,accuracy=1)) + 
  scale_x_discrete(&quot;Maternal species&quot;, guide = guide_axis(angle=90), expand = expansion(mult=0.1))+
  theme(legend.position=&quot;top&quot;)</code></pre>
<p><img src="figures-comp/compare_germination-1.svg" width="768" /></p>
</div>
</div>
<div id="style-length" class="section level1">
<h1>Style length</h1>
<pre class="r"><code>count(styles, momsp, plantid) %&gt;% count(momsp) %&gt;% kable(caption = &quot;Number of plants for style length&quot;)</code></pre>
<table>
<caption>Number of plants for style length</caption>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
<pre class="r"><code>styles.emm &lt;- glmmTMB(style_length_mm ~ momsp + (1|plantid), data=styles) %&gt;% 
  emmeans(~momsp) %&gt;% summary(type=&quot;response&quot;) %&gt;% tidy %&gt;% rename(style_length_mm = estimate) %&gt;% 
  left_join(mating) %&gt;% group_by(pair) %&gt;% 
  mutate(momsp=factor(momsp, levels=sp_pairs), 
         style_diff_mm = style_length_mm-min(style_length_mm),
         style_diff_rel = style_diff_mm / min(style_length_mm))

ggplot(styles, aes(x=momsp, y=style_length_mm, fill=momsp)) + 
  geom_violin(scale=&quot;width&quot;)+ 
  geom_tile(data=tibble(momsp=sp_pairs[c(1,2,5,6,9,10)]), aes(x=momsp, y=4, height=8, width=1),  alpha=0.4, fill=&quot;grey&quot;, inherit.aes = F) +
  geom_violin(scale=&quot;width&quot;, draw_quantiles = c(0.25,0.5,0.75))+ 
  geom_pointrange(data=styles.emm, aes(ymax=conf.high, ymin=conf.low)) + 
  scale_fill_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) + scale_y_continuous(limits=c(0,NA))+
  labs(x=&quot;Species&quot;, y=&quot;Style length (mm)&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/style_length-1.svg" width="576" /></p>
<pre class="r"><code>styles.emm %&gt;%left_join(styles %&gt;% group_by(momsp, plantid) %&gt;% summarize(style_length_mm = mean(style_length_mm)) %&gt;% 
  group_by(momsp) %&gt;% summarize(sd=sd(style_length_mm))) %&gt;% select(momsp, style_length_mm, std.error, sd, style_diff_rel) %&gt;% 
  kable(caption=&quot;comparison of style length relative to short-styled species&quot;)</code></pre>
<table style="width:100%;">
<caption>comparison of style length relative to short-styled
species</caption>
<colgroup>
<col width="7%" />
<col width="20%" />
<col width="23%" />
<col width="14%" />
<col width="10%" />
<col width="22%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">pair</th>
<th align="left">momsp</th>
<th align="right">style_length_mm</th>
<th align="right">std.error</th>
<th align="right">sd</th>
<th align="right">style_diff_rel</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">menziesii</td>
<td align="right">6.595</td>
<td align="right">0.1838</td>
<td align="right">0.7509</td>
<td align="right">0.6634</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">salicaria</td>
<td align="right">3.965</td>
<td align="right">0.1838</td>
<td align="right">0.5266</td>
<td align="right">0.0000</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">lydgatei</td>
<td align="right">3.840</td>
<td align="right">0.1553</td>
<td align="right">0.5169</td>
<td align="right">0.8515</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">sarmentosa</td>
<td align="right">2.074</td>
<td align="right">0.2055</td>
<td align="right">0.1635</td>
<td align="right">0.0000</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">stellarioides</td>
<td align="right">4.498</td>
<td align="right">0.1838</td>
<td align="right">0.4483</td>
<td align="right">0.4662</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">spergulina</td>
<td align="right">3.068</td>
<td align="right">0.1838</td>
<td align="right">0.4707</td>
<td align="right">0.0000</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">membranacea</td>
<td align="right">4.460</td>
<td align="right">0.2055</td>
<td align="right">0.1665</td>
<td align="right">0.0399</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">kauaiensis</td>
<td align="right">4.289</td>
<td align="right">0.1562</td>
<td align="right">0.2922</td>
<td align="right">0.0000</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">kaalae</td>
<td align="right">5.357</td>
<td align="right">0.1838</td>
<td align="right">0.3640</td>
<td align="right">0.2302</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">hookeri</td>
<td align="right">4.354</td>
<td align="right">0.1372</td>
<td align="right">0.4571</td>
<td align="right">0.0000</td>
</tr>
</tbody>
</table>
</div>
<div id="phylogenetic-distance-and-style-length" class="section level1">
<h1>Phylogenetic distance and style length</h1>
<p>Load <em>Schiedea</em> tree inferred with RAxML from Willyard et
al.Â nuclear dataset (no chloroplast data, 1 accession per species).</p>
<pre class="r"><code>tree &lt;- read.nexus(&quot;data/RAxML_1ps_nocp_bipartitionsBranchLabels.sch.short.nobs.nex&quot;)
plot(tree, tip.color = sp_pal[tree$tip.label])</code></pre>
<p><img src="figures-comp/phylodist-1.svg" width="768" /></p>
<pre class="r"><code># par(mfrow=c(1,2))
# par(mar=c(1.5,0,1.5,0))
tree_pairs &lt;- tree %&gt;% keep.tip(sp_pairs)
# plot(tree_pairs, tip.color = sp_pal[tree_pairs$tip.label])
# par(mar=c(0,0,0,0))
# sp_pairs %&gt;% matrix(ncol=2, byrow=T) %&gt;% 
#   arcplot(show.labels=F, horizontal=F, ordering=tree_pairs$tip.label, 
#           col.arcs=sp_pal[(1:5)*2], col.nodes = sp_pal)
# par(mfrow=c(1,1))

hybrid.phylodist &lt;- comp.emm %&gt;% left_join(select(hybrid.test, momsp, dadsp, -type)) %&gt;% 
  mutate(pd = map2_dbl(as.character(momsp), as.character(dadsp), ~cophenetic(tree_pairs)[.x,.y])) %&gt;% 
  select(-df)
phylodist &lt;- hybrid.phylodist %&gt;% group_by(momsp, dadsp) %&gt;% summarize(pd=pd[1])

#TODO use a binomial GLM here, with the raw data?
hybrid.phylodist.test &lt;- hybrid.phylodist %&gt;% left_join(styles.emm, by=&quot;momsp&quot;) %&gt;% group_by(trait, type) %&gt;% nest() %&gt;% 
  mutate(model = map(data, ~lm(estimate ~ pd + style_diff_mm, data=.x)),
         anova = map(model, ~tidy(car::Anova(.x)))) %&gt;% 
  select(-data, -model) %&gt;% unnest(anova) %&gt;% filter(term !=&quot;Residuals&quot;) 

hybrid.phylodist.test %&gt;% select(trait, type, term, p.value) %&gt;% filter(type !=&quot;intra&quot;) %&gt;%
  mutate(trait = propnames[trait], type= typenames[type], 
         term=recode(term, pd=&quot;Phylogenetic distance&quot;, style_diff_mm=&quot;Style length difference&quot;)) %&gt;% 
  pivot_wider(names_from=term, values_from=p.value) %&gt;% 
  kable(caption = &quot;p-values from models of each proportion&quot;, digits=3)</code></pre>
<table>
<caption>p-values from models of each proportion</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="28%" />
<col width="30%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">trait</th>
<th align="left">type</th>
<th align="right">Phylogenetic distance</th>
<th align="right">Style length difference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Capsule formation</td>
<td align="left">Interspecific</td>
<td align="right">0.024</td>
<td align="right">0.555</td>
</tr>
<tr class="even">
<td align="left">Capsule formation</td>
<td align="left">Competitive</td>
<td align="right">0.527</td>
<td align="right">0.686</td>
</tr>
<tr class="odd">
<td align="left">Seed number</td>
<td align="left">Interspecific</td>
<td align="right">0.007</td>
<td align="right">0.772</td>
</tr>
<tr class="even">
<td align="left">Seed number</td>
<td align="left">Competitive</td>
<td align="right">0.128</td>
<td align="right">0.637</td>
</tr>
<tr class="odd">
<td align="left">Germination</td>
<td align="left">Interspecific</td>
<td align="right">0.007</td>
<td align="right">0.432</td>
</tr>
<tr class="even">
<td align="left">Germination</td>
<td align="left">Competitive</td>
<td align="right">0.044</td>
<td align="right">0.139</td>
</tr>
<tr class="odd">
<td align="left">Survival</td>
<td align="left">Interspecific</td>
<td align="right">0.307</td>
<td align="right">0.777</td>
</tr>
<tr class="even">
<td align="left">Survival</td>
<td align="left">Competitive</td>
<td align="right">0.044</td>
<td align="right">0.466</td>
</tr>
<tr class="odd">
<td align="left">Hybrid</td>
<td align="left">Competitive</td>
<td align="right">0.023</td>
<td align="right">0.141</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggplot(left_join(hybrid.phylodist, filter(hybrid.phylodist.test, term==&quot;pd&quot;)), aes(x=pd, y=estimate)) + 
  facet_grid(trait~type, labeller=as_labeller(c(propnames, typenames)), scales=&quot;free_y&quot;)+ coord_cartesian(ylim=c(0,NA))+
  geom_smooth(aes(linetype=paste(p.value &lt; 0.05, type==&quot;intra&quot;)), method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) +
  geom_pointrange(aes(ymax=conf.high, ymin=conf.low, color=momsp)) + scale_color_brewer(palette=&quot;Paired&quot;) + pct_y+ 
  scale_linetype_manual(values=c(2,0,1,0), guide=&quot;none&quot;) + scale_alpha_manual(breaks=c(T,F), values=c(0.5, 1))+
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, color=&quot;Maternal species&quot;) + theme_bw() +
  theme(axis.title.y=element_blank(), axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;,
        panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/phylodist-2.svg" width="768" /></p>
<pre class="r"><code>postzygotic.ratios.phylo &lt;- postzygotic.ratios %&gt;% left_join(comp.tests) %&gt;% left_join(phylodist) %&gt;% 
  mutate(signif = adj.p.value &lt; 0.05) %&gt;% 
  filter(!(trait==&quot;capsule.formed&quot; &amp; momsp %in% c(&quot;spergulina&quot;, &quot;sarmentosa&quot;))) #whole inflor in organza bag - no capsule data

ggplot(postzygotic.ratios.phylo, aes(x=pd, y=RI)) + 
  facet_wrap(vars(trait), labeller=as_labeller(propnames), nrow=1)+ 
  geom_hline(yintercept=0, linetype=2) + geom_hline(yintercept=-1)+ geom_hline(yintercept=1)+
  geom_smooth(method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) + 
  geom_errorbar(aes(color=momsp, ymin=low, ymax=high), width=0.001, linewidth=0.5) +
  geom_point(aes(color=momsp, shape = signif), size=2) + 
  scale_color_brewer(palette=&quot;Paired&quot;) +  scale_shape_manual(values=c(1, 19),guide=&quot;none&quot;)+
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, color=&quot;Maternal species&quot;, y=&quot;Reproductive isolation&quot;) + 
  scale_y_continuous(limits=c(-1,1), breaks=c(-1,-0.5,0,0.5,1), expand=c(0,0),
                     labels=c(&quot;heterosis -1&quot;,&quot;-0.5&quot;,&quot;no barrier 0&quot;,&quot;0.5&quot;,&quot;complete isolation 1&quot;)) + 
  theme_bw() + theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;, 
                     panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/phylodist-3.svg" width="768" /></p>
<pre class="r"><code>postzygotic.ratios.phylo %&gt;% arrange(trait, pd) %&gt;% select(trait, momsp, dadsp, RI, adj.p.value) %&gt;% 
  kable(digits=3, caption = &quot;compare inter vs. intraspecific crosses&quot;)</code></pre>
<table>
<caption>compare inter vs.Â intraspecific crosses</caption>
<thead>
<tr class="header">
<th align="left">trait</th>
<th align="left">momsp</th>
<th align="left">dadsp</th>
<th align="right">RI</th>
<th align="right">adj.p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">menziesii</td>
<td align="left">salicaria</td>
<td align="right">0.047</td>
<td align="right">0.974</td>
</tr>
<tr class="even">
<td align="left">capsule.formed</td>
<td align="left">salicaria</td>
<td align="left">menziesii</td>
<td align="right">0.011</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">lydgatei</td>
<td align="left">sarmentosa</td>
<td align="right">0.004</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">capsule.formed</td>
<td align="left">stellarioides</td>
<td align="left">spergulina</td>
<td align="right">0.015</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">kauaiensis</td>
<td align="left">membranacea</td>
<td align="right">0.537</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">capsule.formed</td>
<td align="left">membranacea</td>
<td align="left">kauaiensis</td>
<td align="right">0.311</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">hookeri</td>
<td align="left">kaalae</td>
<td align="right">0.088</td>
<td align="right">0.932</td>
</tr>
<tr class="even">
<td align="left">capsule.formed</td>
<td align="left">kaalae</td>
<td align="left">hookeri</td>
<td align="right">0.014</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">sds</td>
<td align="left">menziesii</td>
<td align="left">salicaria</td>
<td align="right">0.065</td>
<td align="right">0.040</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">salicaria</td>
<td align="left">menziesii</td>
<td align="right">-0.051</td>
<td align="right">0.224</td>
</tr>
<tr class="odd">
<td align="left">sds</td>
<td align="left">lydgatei</td>
<td align="left">sarmentosa</td>
<td align="right">0.047</td>
<td align="right">0.247</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">sarmentosa</td>
<td align="left">lydgatei</td>
<td align="right">0.071</td>
<td align="right">0.101</td>
</tr>
<tr class="odd">
<td align="left">sds</td>
<td align="left">spergulina</td>
<td align="left">stellarioides</td>
<td align="right">0.017</td>
<td align="right">0.635</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">stellarioides</td>
<td align="left">spergulina</td>
<td align="right">0.184</td>
<td align="right">0.058</td>
</tr>
<tr class="odd">
<td align="left">sds</td>
<td align="left">kauaiensis</td>
<td align="left">membranacea</td>
<td align="right">0.434</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">membranacea</td>
<td align="left">kauaiensis</td>
<td align="right">0.350</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">sds</td>
<td align="left">hookeri</td>
<td align="left">kaalae</td>
<td align="right">0.336</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">kaalae</td>
<td align="left">hookeri</td>
<td align="right">0.104</td>
<td align="right">0.224</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">menziesii</td>
<td align="left">salicaria</td>
<td align="right">-0.187</td>
<td align="right">0.011</td>
</tr>
<tr class="even">
<td align="left">prop.germ</td>
<td align="left">salicaria</td>
<td align="left">menziesii</td>
<td align="right">-0.193</td>
<td align="right">0.002</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">lydgatei</td>
<td align="left">sarmentosa</td>
<td align="right">0.002</td>
<td align="right">0.978</td>
</tr>
<tr class="even">
<td align="left">prop.germ</td>
<td align="left">sarmentosa</td>
<td align="left">lydgatei</td>
<td align="right">-0.008</td>
<td align="right">0.978</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">spergulina</td>
<td align="left">stellarioides</td>
<td align="right">0.063</td>
<td align="right">0.620</td>
</tr>
<tr class="even">
<td align="left">prop.germ</td>
<td align="left">stellarioides</td>
<td align="left">spergulina</td>
<td align="right">0.059</td>
<td align="right">0.620</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">kauaiensis</td>
<td align="left">membranacea</td>
<td align="right">0.262</td>
<td align="right">0.620</td>
</tr>
<tr class="even">
<td align="left">prop.germ</td>
<td align="left">membranacea</td>
<td align="left">kauaiensis</td>
<td align="right">0.877</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">hookeri</td>
<td align="left">kaalae</td>
<td align="right">-0.050</td>
<td align="right">0.407</td>
</tr>
<tr class="even">
<td align="left">prop.germ</td>
<td align="left">kaalae</td>
<td align="left">hookeri</td>
<td align="right">0.616</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">menziesii</td>
<td align="left">salicaria</td>
<td align="right">-0.644</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">salicaria</td>
<td align="left">menziesii</td>
<td align="right">-0.375</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">lydgatei</td>
<td align="left">sarmentosa</td>
<td align="right">-0.260</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">sarmentosa</td>
<td align="left">lydgatei</td>
<td align="right">-0.514</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">spergulina</td>
<td align="left">stellarioides</td>
<td align="right">-0.532</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">stellarioides</td>
<td align="left">spergulina</td>
<td align="right">-0.056</td>
<td align="right">0.122</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">kauaiensis</td>
<td align="left">membranacea</td>
<td align="right">0.025</td>
<td align="right">0.822</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">membranacea</td>
<td align="left">kauaiensis</td>
<td align="right">0.158</td>
<td align="right">0.333</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">hookeri</td>
<td align="left">kaalae</td>
<td align="right">-0.023</td>
<td align="right">0.283</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">kaalae</td>
<td align="left">hookeri</td>
<td align="right">0.056</td>
<td align="right">0.094</td>
</tr>
</tbody>
</table>
<pre class="r"><code>postzygotic.ratios.phylo %&gt;% group_by(trait) %&gt;% nest() %&gt;% 
  mutate(model = map(data, ~lm(RI ~ pd, data=.x)), 
         coef = map(model, tidy)) %&gt;% 
  select(trait, coef) %&gt;% unnest(coef) %&gt;% filter(term != &quot;(Intercept)&quot;) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">trait</th>
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">pd</td>
<td align="right">15.23</td>
<td align="right">8.916</td>
<td align="right">1.708</td>
<td align="right">0.1385</td>
</tr>
<tr class="even">
<td align="left">sds</td>
<td align="left">pd</td>
<td align="right">18.95</td>
<td align="right">5.015</td>
<td align="right">3.779</td>
<td align="right">0.0054</td>
</tr>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">pd</td>
<td align="right">36.17</td>
<td align="right">12.447</td>
<td align="right">2.906</td>
<td align="right">0.0197</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">pd</td>
<td align="right">34.77</td>
<td align="right">7.913</td>
<td align="right">4.394</td>
<td align="right">0.0023</td>
</tr>
</tbody>
</table>
<div id="seed-size" class="section level2">
<h2>Seed size</h2>
<pre class="r"><code>postzygotic.ratios.phylo %&gt;% left_join(seedsize) %&gt;% filter(trait %in% c(&quot;prop.germ&quot;, &quot;prop.alive&quot;)) %&gt;% 
  ggplot(aes(x=seedlength_mm, y=RI)) + 
  facet_wrap(vars(trait), labeller=as_labeller(propnames), nrow=1)+ 
  geom_hline(yintercept=0, linetype=2) + geom_hline(yintercept=-1)+ geom_hline(yintercept=1)+
  geom_errorbar(aes(color=momsp, ymin=low, ymax=high), width=0.001, linewidth=0.5) +
  geom_path(aes(group=pair, color=momsp))+
  geom_point(aes(color=momsp, shape = signif), size=2) + 
  scale_color_brewer(palette=&quot;Paired&quot;) +  scale_shape_manual(values=c(1, 19),guide=&quot;none&quot;)+
  labs(x=&quot;Seed length (mm)&quot;, color=&quot;Maternal species&quot;, y=&quot;Reproductive isolation&quot;) + 
  scale_y_continuous(limits=c(-1,1), breaks=c(-1,-0.5,0,0.5,1), expand=c(0,0),
                     labels=c(&quot;heterosis -1&quot;,&quot;-0.5&quot;,&quot;no barrier 0&quot;,&quot;0.5&quot;,&quot;complete isolation 1&quot;)) + 
  theme_bw() + theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;, 
                     panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/seedsize-1.svg" width="672" /></p>
</div>
<div id="competitive-pollinations-and-phylogeny" class="section level2">
<h2>Competitive pollinations and phylogeny</h2>
<pre class="r"><code>mating.tree &lt;- data.frame(momsp=tree_pairs$tip.label) %&gt;% left_join(mating)
layout(matrix(c(1,2), nrow=1), widths=c(4,1))
par(mar=c(0,0,0,0))
plot.phylo(tree_pairs, tip.color = sp_pal[tree_pairs$tip.label], edge.width=5, edge.col =&quot;grey&quot;, label.offset=0.0002, font=4, cex=1.2,  x.lim=0.025)
add.scale.bar(x=0,y=10, lwd=5, lcol=&quot;grey&quot;, col=&quot;grey&quot;)
text(x=0.025, y=1:10, mating.tree$mating, cex=1.2, col=sp_pal[tree_pairs$tip.label]) #c(&quot;H&quot;=&quot;\U26A5&quot;,&quot;D&quot;=&quot;\U2640\U2642&quot;,&quot;G&quot;=&quot;\U26A5\U2640&quot;)
ntaxa &lt;- Ntip(tree_pairs)
sp_pairs %&gt;% matrix(ncol=2, byrow=T) %&gt;% 
  arcplot(show.labels=F, horizontal=F, ordering=tree_pairs$tip.label, 
          col.arcs=sp_pal[(1:5)*2], col.nodes = sp_pal, lwd.arcs=5, 
          ylim = range(arcdiagram::xynodes(ntaxa, 1:ntaxa, labels = 1:ntaxa)))</code></pre>
<p><img src="figures-comp/phylo_arc-1.svg" width="480" /></p>
<pre class="r"><code>left_join(hybrid.phylodist, filter(hybrid.phylodist.test, term==&quot;pd&quot;)) %&gt;% 
  filter(trait==&quot;prop.hybrid&quot;, type==&quot;mixed&quot;) %&gt;% left_join(hybrid.test) %&gt;% 
  ggplot(aes(x=pd, y=estimate)) + coord_cartesian(xlim=c(0,NA))+
  geom_hline(yintercept=0.5, linetype=2)+
  geom_smooth(method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) +
  geom_pointrange(aes(ymax=conf.high, ymin=conf.low, color=momsp, shape=p.adj &lt; 0.05)) + 
  scale_shape_manual(&quot;Different from 50%&quot;, values=c(1, 19), labels=c(&quot;no&quot;,&quot;yes&quot;))+
  scale_color_brewer(palette=&quot;Paired&quot;) + 
  scale_alpha_manual(breaks=c(T,F), values=c(0.5, 1))+
  scale_y_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0.02,0), limits=c(0,NA))+
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, color=&quot;Maternal species&quot;,
       y=&quot;Hybrids from competitive pollinations&quot;) + theme_minimal() +
  theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;, legend.box = &quot;vertical&quot;,
        panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/phylodist_hybrid-1.svg" width="672" /></p>
<pre class="r"><code>hybrid.phylodist %&gt;% filter(trait==&quot;prop.hybrid&quot;, type==&quot;mixed&quot;) %&gt;% 
  left_join(select(hybrid.test, momsp, p.adj)) %&gt;% 
  left_join(select(styles.emm, momsp, style_length_mm, style.se = std.error)) %&gt;% 
  pivot_longer(c(pd, style_length_mm)) %&gt;% 
  {
    ggplot(., aes(x=value, y=estimate, color=momsp)) + 
      facet_wrap(vars(name), scales=&quot;free_x&quot;, strip.position = &quot;bottom&quot;,
                 labeller=as_labeller(c(pd=&quot;Phylogenetic distance (mutations/site)&quot;, style_length_mm = &quot;Style length (mm)&quot;)))+
      geom_hline(yintercept=0.5, linetype=2)+
      geom_smooth(data = filter(., name==&quot;pd&quot;), method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) +
      geom_line(data = filter(., name==&quot;style_length_mm&quot;), aes(group=pair))+
      geom_errorbarh(data = filter(., name==&quot;style_length_mm&quot;), aes(xmin=value-style.se, xmax=value+style.se))+
      geom_pointrange(aes(ymax=estimate+std.error, ymin=estimate-std.error, shape=p.adj &lt; 0.05)) + 
      scale_shape_manual(values=c(1, 19), guide=&quot;none&quot;)+
      scale_color_brewer(palette=&quot;Paired&quot;) + 
      scale_alpha_manual(breaks=c(T,F), values=c(0.5, 1))+
      scale_y_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0.02,0), limits=c(0,NA))+
      labs(color=&quot;Maternal species&quot;, y=&quot;Hybrids from competitive pollinations&quot;) + theme_minimal() +
      theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;,
            axis.title.x=element_blank(), panel.grid.minor = element_blank(),  panel.spacing = unit(1, &quot;lines&quot;), 
            strip.placement = &quot;outside&quot;, strip.text = element_text(size=11)) 
  }</code></pre>
<p><img src="figures-comp/phylodist_hybrid-2.svg" width="672" /></p>
<p>Regressions against the phylogenetic distance of each species pair.
Solid regression lines are statistically significant in a model that
also included the difference in style length within pairs. Note that for
reference, data for intraspecific crosses is plotted with the
phylogenetic distance to its paired species, which doesnâ€™t make great
sense.</p>
<pre class="r"><code>hybrid.phylodist %&gt;% filter(trait==&quot;prop.hybrid&quot;,  type==&quot;mixed&quot;) %&gt;% 
  left_join(select(styles.emm, momsp, style_length_mm, mating, se=std.error)) %&gt;% 
  ggplot(aes(x=pd, y=style_length_mm, color=momsp, shape=mating)) + 
  geom_pointrange(aes(ymin=style_length_mm-se, ymax=style_length_mm+se)) + 
  scale_color_brewer(palette=&quot;Paired&quot;) + coord_cartesian(xlim=c(0, NA), ylim=c(0,NA)) +
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, y=&quot;Style length (mm)&quot;, color=&quot;Species&quot;, shape=&quot;Mating&quot;) + 
  theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figures-comp/styles-1.svg" width="768" /></p>
<p>The range of phylogenetic distances and style lengths in the study.
The species in each pair with a longer style is given a lighter
color.</p>
</div>
<div id="hybrid-seed-ratios" class="section level2">
<h2>Hybrid seed ratios</h2>
<p>To estimate the proportion of viable seeds sired by heterospecific
pollen, divide the surviving hybrid ratios in mixed crosses by the
â€œhybrid vigorâ€: the germination and survival of hybrid seeds relative to
conspecific seeds.</p>
<pre class="r"><code>hybrid.phylostyle &lt;- left_join(hybrid.test, phylodist) %&gt;% 
  left_join(select(styles.emm, momsp, style_length_mm, style_diff_rel)) %&gt;% 
  mutate(hybrid.RI.all = 1-2*hybrid.vigor.all/(hybrid.vigor.all+1)) %&gt;% #RI per Sobel and Chen 2014 
  pivot_longer(all_of(c(names(propnames[c(1,4)]), &quot;hybrid.RI.all&quot;)) )

hybrid.phylostyle %&gt;% pivot_longer(c(pd, style_length_mm), names_to = &quot;xname&quot;, values_to = &quot;xvalue&quot;) %&gt;% 
  ggplot(aes(x=xvalue, y=value)) + 
  facet_grid(name ~ xname, scales=&quot;free&quot;, switch=&quot;both&quot;, 
             labeller=as_labeller(c(hybrid.RI.all=&quot;Reproductive isolation\n(heterosis \U2194 complete isolation)&quot;,
                                    prop.hybrid=&quot;Proportion of hybrid plants\n(observed)&quot;,
                                    prop.hybrid.zyg = &quot;Hybrids pollen competiveness\n(estimated)&quot;,
                                    style_length_mm = &quot;Style length (mm)&quot;, 
                                    pd=&quot;Phylogenetic distance (mutations/site)&quot;))) + 
  geom_hline(aes(yintercept=if_else(name==&quot;hybrid.RI.all&quot;,0,0.5)), linetype=2)+
  geom_smooth(aes(linetype=if_else(xname==&quot;pd&quot;,&quot;yes&quot;,&quot;no&quot;)), method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) +
  geom_path(aes(linetype=if_else(xname==&quot;style_length_mm&quot; &amp; name != &quot;hybrid.vigor.germsurv&quot;,&quot;yes&quot;,&quot;no&quot;), 
                group=pair, color=sp_pairs[as.integer(pair)*2]), show.legend = F)+
  geom_point(aes(color=momsp), size=4) + 
  scale_linetype_manual(values=c(yes=&quot;solid&quot;, no=&quot;blank&quot;), guide=&quot;none&quot;) +
  scale_y_continuous(expand=expansion(mult=c(0.05,0.05)), breaks=(-2:2)/2) + 
  expand_limits(y=c(0,0.6)) +
  scale_color_brewer(&quot;Maternal\nspecies&quot;, palette=&quot;Paired&quot;) + theme_minimal()+
  theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;, panel.background = element_rect(),
        strip.placement = &quot;outside&quot;, axis.title.x = element_blank(), axis.title.y = element_blank(),
        panel.spacing.y = unit(0.5,&quot;lines&quot;), panel.spacing.x = unit(1,&quot;lines&quot;), panel.grid = element_blank()) </code></pre>
<p><img src="figures-comp/fertilized-1.svg" width="633.6" /></p>
<pre class="r"><code>hybrid.phylostyle %&gt;% group_by(name) %&gt;% nest() %&gt;% 
  mutate(model = map(data, ~lm(value ~ pd + style_diff_rel, data=.x)), 
         coef = map(model, tidy)) %&gt;% 
  select(name, coef) %&gt;% unnest(coef) %&gt;% filter(term != &quot;(Intercept)&quot;) %&gt;% kable()</code></pre>
<table>
<colgroup>
<col width="22%" />
<col width="22%" />
<col width="14%" />
<col width="14%" />
<col width="14%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">name</th>
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">capsule.formed</td>
<td align="left">pd</td>
<td align="right">-20.4473</td>
<td align="right">11.3916</td>
<td align="right">-1.7950</td>
<td align="right">0.1157</td>
</tr>
<tr class="even">
<td align="left">capsule.formed</td>
<td align="left">style_diff_rel</td>
<td align="right">0.0598</td>
<td align="right">0.2446</td>
<td align="right">0.2444</td>
<td align="right">0.8139</td>
</tr>
<tr class="odd">
<td align="left">prop.alive</td>
<td align="left">pd</td>
<td align="right">-149.7522</td>
<td align="right">54.2623</td>
<td align="right">-2.7598</td>
<td align="right">0.0281</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">style_diff_rel</td>
<td align="right">-0.5421</td>
<td align="right">1.1650</td>
<td align="right">-0.4653</td>
<td align="right">0.6558</td>
</tr>
<tr class="odd">
<td align="left">hybrid.RI.all</td>
<td align="left">pd</td>
<td align="right">90.8415</td>
<td align="right">14.6054</td>
<td align="right">6.2197</td>
<td align="right">0.0004</td>
</tr>
<tr class="even">
<td align="left">hybrid.RI.all</td>
<td align="left">style_diff_rel</td>
<td align="right">0.3407</td>
<td align="right">0.3136</td>
<td align="right">1.0867</td>
<td align="right">0.3132</td>
</tr>
</tbody>
</table>
<pre class="r"><code>left_join(filter(comp.emm, trait==&quot;prop.hybrid&quot;), hybrid.test) %&gt;% 
left_join(select(styles.emm, momsp, style_length_mm, mating, style_conf.low=conf.low, style_conf.high=conf.high)) %&gt;% 
  ggplot(aes(x=style_length_mm, y=estimate, color=momsp)) + geom_hline(aes(yintercept=0.5), linetype=2) +  
  geom_errorbarh(aes(xmin=style_conf.low, xmax=style_conf.high)) +  geom_errorbar(aes(ymin=conf.low, ymax=conf.high)) +
  geom_line(aes(group=pair))+  geom_point(size=4)+
  scale_color_brewer(palette=&quot;Paired&quot;) + scale_y_continuous(labels=scales::percent_format(), breaks=(0:4)/4)+ scale_x_continuous(limits=c(0,NA))+
  labs(x=&quot;Style length (mm)&quot;, color=&quot;Maternal\nspecies&quot;, y=&quot;Hybrids from competitive pollinations&quot;) + 
  theme_minimal() + theme(axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;,
                          legend.key.size=unit(0.8, &quot;lines&quot;))+
  guides(color = guide_legend(override.aes = list(size = 4)))</code></pre>
<p><img src="figures-comp/progeny_styles-1.svg" width="576" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
