<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="John Powers" />

<meta name="date" content="2023-04-04" />

<title>Schiedea pollen competition</title>

<script src="libs/header-attrs-2.20/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<script src="libs/navigation-1.1/codefolding.js"></script>
<link href="libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore"><em>Schiedea</em> pollen competition</h1>
<h4 class="author"><a href="http://sites.uci.edu/powers/">John
Powers</a></h4>
<h4 class="date">2023-04-04</h4>

</div>


<pre class="r"><code>library(tidyverse)
library(glmmTMB)
library(broom)
library(emmeans)
library(knitr)
knitr::opts_chunk$set(comment=&quot;&quot;, cache=T, warning = F, message = F, fig.height=8, fig.width=8,
                      fig.path = &quot;figures-comp/&quot;, dev = &quot;svglite&quot;, dev.args=list(fix_text_size=FALSE))
options(digits=4, knitr.kable.NA = &quot;&quot;)
ggplot2::theme_set(theme_minimal())</code></pre>
<div id="inventory" class="section level1">
<h1>Inventory</h1>
<pre class="r"><code>#in order of phylogenetic distance, pair order is by style length
sp_pairs &lt;- c(&quot;menz&quot;=&quot;menziesii&quot;, &quot;sal&quot;=&quot;salicaria&quot;,
              &quot;lydg&quot;=&quot;lydgatei&quot;, &quot;sarm&quot; =&quot;sarmentosa&quot;,
              &quot;stell&quot;=&quot;stellarioides&quot;,&quot;sperg&quot;=&quot;spergulina&quot;,
              &quot;memb&quot;=&quot;membranacea&quot;,&quot;kaua&quot;=&quot;kauaiensis&quot;,
              &quot;kaal&quot;=&quot;kaalae&quot;, &quot;hook&quot;=&quot;hookeri&quot;)
typenames &lt;- set_names(c(&quot;Intraspecific&quot;,&quot;Competitive&quot;,&quot;Interspecific&quot;),c(&quot;intra&quot;,&quot;mixed&quot;,&quot;inter&quot;))
type_pal &lt;- set_names(c(&quot;#DC0060&quot;, &quot;#86D400&quot;, &quot;#F56E00&quot;), names(typenames))
propnames &lt;- set_names(c(&quot;Germinated&quot;,&quot;Survived&quot;,&quot;Hybrid&quot;), c(&quot;prop.germ&quot;, &quot;prop.alive&quot;, &quot;prop.hybrid&quot;))

momid.fixes &lt;- read_tsv(&quot;data/Pollen competition - momid_fixes.tsv&quot;)
momid.fixes.germ &lt;- momid.fixes %&gt;% drop_na(germid, momid_fixed) %&gt;% select(momid, momid_fixed) %&gt;% deframe()
momid.fixes.surv &lt;- momid.fixes %&gt;% drop_na(survid, momid_fixed) %&gt;% select(momid, momid_fixed) %&gt;% deframe()

germ.pot &lt;- read_tsv(&quot;data/Pollen competition - germination.tsv&quot;) %&gt;%  
  mutate(dadspm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp1, dadsp2), #rearrange dad1/dad2 to have dadsp match momsp
         dadidm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid1, dadid2),
         dadsp  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp2, dadsp1),
         dadid  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid2, dadid1), .after=momid) %&gt;% 
  select(-starts_with(&quot;germinated_n&quot;), -transplanted_n, -matches(&quot;dad(sp|id)[12]&quot;)) %&gt;% 
  drop_na(germinated) %&gt;% #exclude one contaminated cross and &quot;intra inv&quot; inviability tests
  mutate(across(matches(&quot;sp&quot;), ~ recode(.x, !!!sp_pairs)),
         across(matches(&quot;(dad|mom)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)), 
         across(matches(&quot;(dad|mom)id&quot;), ~recode(.x, !!!momid.fixes.germ)), #TODO are the momid fixes applicable to dads too?
         sxc = paste(momsp, type, sep=&quot;.&quot;),
         #across(where(is.character), factor), 
         type=factor(type, levels= names(typenames)), 
         across(matches(&quot;sp&quot;), ~ factor(.x, levels = sp_pairs)),
         germinated = ifelse(germinated &gt; planted, planted, germinated))

germ &lt;- germ.pot %&gt;% group_by(date_planted, type, momsp, sxc, momid, dadspm, dadidm, dadsp, dadid) %&gt;% #date_planted splits 30 plant x plant crosses
  summarize(planted = sum(planted), germinated = sum(germinated), .groups=&quot;drop&quot;) %&gt;% # add together all the pots of one cross
  mutate(prop.germ = germinated/planted) # proportion of planted seeds that germinated 

surv &lt;- read_tsv(&quot;data/Pollen competition - survival.tsv&quot;) %&gt;% 
  mutate(dadspm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp1, dadsp2), #rearrange dad1/dad2 to have dadsp match momsp
         dadidm = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid1, dadid2),
         dadsp  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadsp2, dadsp1),
         dadid  = if_else(type==&quot;mixed&quot; &amp; dadsp1==momsp, dadid2, dadid1), .after=momid) %&gt;% 
  select(-matches(&quot;dad(sp|id)[12]&quot;)) %&gt;% drop_na(dead) %&gt;% #one row excluded
  mutate(across(matches(&quot;(mom|dad)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)), 
         across(matches(&quot;(dad|mom)id&quot;), ~recode(.x, !!!momid.fixes.surv)),
         sxc = paste(momsp, type, sep=&quot;.&quot;),
         #across(where(is.character), factor), 
         type=factor(type, levels= names(typenames)),
         across(matches(&quot;sp&quot;), ~ factor(.x, levels = sp_pairs)),
         across(all_of(c(&quot;sameasmom&quot;, &quot;hybrid&quot;)), ~replace_na(.x,0)), 
         alive = sameasmom + hybrid,
         total = alive + dead, 
         prop.hybrid = hybrid / alive,
         prop.alive = alive / total)

surv %&gt;% count(momsp, type, dadsp, name=&quot;crosses_surv&quot;) %&gt;% 
  full_join(germ %&gt;% count(momsp, type, dadsp, name=&quot;crosses_germ&quot;)) %&gt;% 
  full_join(germ.pot %&gt;% count(momsp, type, dadsp, name = &quot;pots_germ&quot;)) %&gt;%
  mutate(pots_per_cross = pots_germ/crosses_germ) %&gt;% 
  kable(caption=&quot;Number of crosses between individual plants used for germination and survival,
        and how many pots the seeds filled&quot;)</code></pre>
<table>
<caption>Number of crosses between individual plants used for
germination and survival, and how many pots the seeds filled</caption>
<colgroup>
<col width="16%" />
<col width="7%" />
<col width="16%" />
<col width="15%" />
<col width="15%" />
<col width="11%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="left">type</th>
<th align="left">dadsp</th>
<th align="right">crosses_surv</th>
<th align="right">crosses_germ</th>
<th align="right">pots_germ</th>
<th align="right">pots_per_cross</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">intra</td>
<td align="left">menziesii</td>
<td align="right">7</td>
<td align="right">8</td>
<td align="right">74</td>
<td align="right">9.250</td>
</tr>
<tr class="even">
<td align="left">menziesii</td>
<td align="left">mixed</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">16</td>
<td align="right">80</td>
<td align="right">5.000</td>
</tr>
<tr class="odd">
<td align="left">menziesii</td>
<td align="left">inter</td>
<td align="left">salicaria</td>
<td align="right">19</td>
<td align="right">18</td>
<td align="right">87</td>
<td align="right">4.833</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">intra</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">97</td>
<td align="right">8.818</td>
</tr>
<tr class="odd">
<td align="left">salicaria</td>
<td align="left">mixed</td>
<td align="left">menziesii</td>
<td align="right">14</td>
<td align="right">12</td>
<td align="right">76</td>
<td align="right">6.333</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="left">inter</td>
<td align="left">menziesii</td>
<td align="right">13</td>
<td align="right">12</td>
<td align="right">53</td>
<td align="right">4.417</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">intra</td>
<td align="left">lydgatei</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">42</td>
<td align="right">8.400</td>
</tr>
<tr class="even">
<td align="left">lydgatei</td>
<td align="left">mixed</td>
<td align="left">sarmentosa</td>
<td align="right">5</td>
<td align="right">11</td>
<td align="right">56</td>
<td align="right">5.091</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="left">inter</td>
<td align="left">sarmentosa</td>
<td align="right">19</td>
<td align="right">20</td>
<td align="right">90</td>
<td align="right">4.500</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">intra</td>
<td align="left">sarmentosa</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">20</td>
<td align="right">10.000</td>
</tr>
<tr class="odd">
<td align="left">sarmentosa</td>
<td align="left">mixed</td>
<td align="left">lydgatei</td>
<td align="right">5</td>
<td align="right">12</td>
<td align="right">63</td>
<td align="right">5.250</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="left">inter</td>
<td align="left">lydgatei</td>
<td align="right">7</td>
<td align="right">9</td>
<td align="right">41</td>
<td align="right">4.556</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">intra</td>
<td align="left">stellarioides</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">26</td>
<td align="right">4.333</td>
</tr>
<tr class="even">
<td align="left">stellarioides</td>
<td align="left">mixed</td>
<td align="left">spergulina</td>
<td align="right">6</td>
<td align="right">6</td>
<td align="right">27</td>
<td align="right">4.500</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="left">inter</td>
<td align="left">spergulina</td>
<td align="right">10</td>
<td align="right">7</td>
<td align="right">13</td>
<td align="right">1.857</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">intra</td>
<td align="left">spergulina</td>
<td align="right">5</td>
<td align="right">9</td>
<td align="right">67</td>
<td align="right">7.444</td>
</tr>
<tr class="odd">
<td align="left">spergulina</td>
<td align="left">mixed</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right">8</td>
<td align="right">50</td>
<td align="right">6.250</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="left">inter</td>
<td align="left">stellarioides</td>
<td align="right">4</td>
<td align="right">3</td>
<td align="right">14</td>
<td align="right">4.667</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">intra</td>
<td align="left">membranacea</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">27</td>
<td align="right">5.400</td>
</tr>
<tr class="even">
<td align="left">membranacea</td>
<td align="left">mixed</td>
<td align="left">kauaiensis</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">60</td>
<td align="right">10.000</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="left">inter</td>
<td align="left">kauaiensis</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">18</td>
<td align="right">4.500</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">intra</td>
<td align="left">kauaiensis</td>
<td align="right">2</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">2.111</td>
</tr>
<tr class="odd">
<td align="left">kauaiensis</td>
<td align="left">mixed</td>
<td align="left">membranacea</td>
<td align="right">7</td>
<td align="right">8</td>
<td align="right">63</td>
<td align="right">7.875</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="left">inter</td>
<td align="left">membranacea</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">3.000</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">intra</td>
<td align="left">kaalae</td>
<td align="right">7</td>
<td align="right">20</td>
<td align="right">35</td>
<td align="right">1.750</td>
</tr>
<tr class="even">
<td align="left">kaalae</td>
<td align="left">mixed</td>
<td align="left">hookeri</td>
<td align="right">10</td>
<td align="right">6</td>
<td align="right">34</td>
<td align="right">5.667</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="left">inter</td>
<td align="left">hookeri</td>
<td align="right">11</td>
<td align="right">73</td>
<td align="right">118</td>
<td align="right">1.616</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">intra</td>
<td align="left">hookeri</td>
<td align="right">12</td>
<td align="right">21</td>
<td align="right">128</td>
<td align="right">6.095</td>
</tr>
<tr class="odd">
<td align="left">hookeri</td>
<td align="left">mixed</td>
<td align="left">kaalae</td>
<td align="right">11</td>
<td align="right">8</td>
<td align="right">70</td>
<td align="right">8.750</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="left">inter</td>
<td align="left">kaalae</td>
<td align="right">34</td>
<td align="right">61</td>
<td align="right">107</td>
<td align="right">1.754</td>
</tr>
</tbody>
</table>
<pre class="r"><code># to run this, comment out across(matches(&quot;(mom|dad)id&quot;), ~str_remove_all(.x, &quot;[ -]&quot;)) lines above
# left_join(count(germ, momsp, germid=momid, name=&quot;germ&quot;) %&gt;% mutate(momid=str_remove_all(germid, &quot;[ -]&quot;)),
#           germ %&gt;% group_by(momsp, germid=momid) %&gt;% summarize(tot.germ=sum(germinated))) %&gt;%
#   full_join(count(surv, momsp, survid=momid, name=&quot;surv&quot;)%&gt;% mutate(momid=str_remove_all(survid, &quot;[ -]&quot;))) %&gt;%
#   arrange(momsp, momid) %&gt;% select(momsp, momid, germid, survid, tot.germ, germ, surv) %&gt;%
#   add_count(momid) 
# #  write_tsv(&quot;data/survgerm_momids.tsv&quot;, na=&quot;&quot;)

mating &lt;- read_tsv(&quot;data/Pollen competition - mating.tsv&quot;) %&gt;% 
  left_join(tibble(momsp=sp_pairs, pair = factor(rep(1:5, each=2)))) %&gt;% 
  mutate(momsp = factor(momsp, level=sp_pairs)) 

styles &lt;- read_tsv(&quot;data/Pollen competition - style_length.tsv&quot;) %&gt;% 
  mutate(momsp = factor(sp_pairs[momsp], levels=sp_pairs))</code></pre>
</div>
<div id="scoring-dates" class="section level1">
<h1>Scoring dates</h1>
<p>The date scored is not always available or accurate, do not use for
analysis.</p>
<pre class="r"><code>germ.pot %&gt;%  ggplot(aes(x=date_planted, y=date_scored, color=momsp)) + 
  geom_abline(slope=1, intercept=0) + geom_abline(slope=1, intercept=90) + geom_point() + geom_rug() +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(title=&quot;Germination scoring&quot;) #TODO check mis-entered dates</code></pre>
<p><img src="figures-comp/dates-1.svg" width="768" /></p>
<pre class="r"><code>germ.pot %&gt;% filter(abs(date_scored-date_planted)&lt;=90) %&gt;% ggplot(aes(x=date_scored - date_planted, fill=type)) +
  facet_wrap(vars(momsp)) +
  geom_histogram(binwidth=1) + scale_x_continuous(limits=c(0,NA)) + 
  labs(title=&quot;Germination scoring&quot;, subtitle = &quot;plot restricted to 90 days&quot;)</code></pre>
<p><img src="figures-comp/dates-2.svg" width="768" /></p>
<pre class="r"><code>surv %&gt;% ggplot(aes(x=date_planted, y=date_scored, color=momsp)) + geom_point() + geom_rug() + 
   scale_color_brewer(palette=&quot;Paired&quot;) + labs(title=&quot;Survival scoring&quot;)</code></pre>
<p><img src="figures-comp/dates-3.svg" width="768" /></p>
</div>
<div id="histograms" class="section level1">
<h1>Histograms</h1>
<pre class="r"><code>ggplot(germ, aes(x=planted, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-1.svg" width="768" /></p>
<pre class="r"><code>ggplot(germ, aes(x=germinated, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-2.svg" width="768" /></p>
<pre class="r"><code>ggplot(filter(surv, type !=&quot;inter&quot;), aes(x=sameasmom, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp))</code></pre>
<p><img src="figures-comp/histogram-3.svg" width="768" /></p>
<pre class="r"><code>ggplot(filter(surv, type !=&quot;intra&quot;), aes(x=hybrid, fill=type)) + geom_histogram(binwidth=1) + facet_wrap(vars(momsp))#lots of 20s</code></pre>
<p><img src="figures-comp/histogram-4.svg" width="768" /></p>
<pre class="r"><code>ggplot(surv, aes(x=dead, fill=type)) + geom_histogram(binwidth=10) + facet_wrap(vars(momsp), scales=&quot;free_y&quot;)</code></pre>
<p><img src="figures-comp/histogram-5.svg" width="768" /></p>
<pre class="r"><code>pct_y &lt;- scale_y_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0,0))
pct_x &lt;- scale_x_continuous(breaks=(0:4)/4, labels=~scales::percent(.,accuracy=1), expand=c(0,0))
germ.surv &lt;- full_join(germ, surv) #TODO fix matching
germ.surv %&gt;% mutate(germalive = paste(is.na(germinated), is.na(alive)) %&gt;% 
                       recode(`FALSE FALSE`=&quot;both&quot;, `FALSE TRUE`=&quot;survival&quot;, `TRUE FALSE`=&quot;germination&quot;)) %&gt;% 
  count(type, momsp, germalive) %&gt;% pivot_wider(names_from=&quot;germalive&quot;, values_from=&quot;n&quot;) %&gt;% 
  kable(kable=&quot;Matching crosses between germination and survival datasets&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="left">momsp</th>
<th align="right">both</th>
<th align="right">survival</th>
<th align="right">germination</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">intra</td>
<td align="left">menziesii</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">intra</td>
<td align="left">salicaria</td>
<td align="right">6</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">intra</td>
<td align="left">lydgatei</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">intra</td>
<td align="left">sarmentosa</td>
<td align="right">2</td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td align="left">intra</td>
<td align="left">stellarioides</td>
<td align="right">5</td>
<td align="right">1</td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">intra</td>
<td align="left">spergulina</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">intra</td>
<td align="left">membranacea</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">intra</td>
<td align="left">kauaiensis</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">intra</td>
<td align="left">kaalae</td>
<td align="right">7</td>
<td align="right">13</td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">intra</td>
<td align="left">hookeri</td>
<td align="right">11</td>
<td align="right">10</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">mixed</td>
<td align="left">menziesii</td>
<td align="right">8</td>
<td align="right">8</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="left">salicaria</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">mixed</td>
<td align="left">lydgatei</td>
<td align="right"></td>
<td align="right">11</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="left">sarmentosa</td>
<td align="right"></td>
<td align="right">12</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">mixed</td>
<td align="left">stellarioides</td>
<td align="right"></td>
<td align="right">6</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="left">spergulina</td>
<td align="right"></td>
<td align="right">8</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">mixed</td>
<td align="left">membranacea</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="left">kauaiensis</td>
<td align="right">6</td>
<td align="right">2</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">mixed</td>
<td align="left">kaalae</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="left">hookeri</td>
<td align="right"></td>
<td align="right">8</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="left">menziesii</td>
<td align="right">16</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">inter</td>
<td align="left">salicaria</td>
<td align="right">10</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="left">lydgatei</td>
<td align="right">9</td>
<td align="right">11</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">inter</td>
<td align="left">sarmentosa</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="left">stellarioides</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">inter</td>
<td align="left">spergulina</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="left">membranacea</td>
<td align="right"></td>
<td align="right">4</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">inter</td>
<td align="left">kauaiensis</td>
<td align="right"></td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">inter</td>
<td align="left">kaalae</td>
<td align="right">11</td>
<td align="right">62</td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">inter</td>
<td align="left">hookeri</td>
<td align="right">34</td>
<td align="right">27</td>
<td align="right"></td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggplot(germ.surv, aes(x=prop.germ, y=prop.alive, color=momsp)) + 
  geom_point() + geom_smooth(se=F, span=1.1) + pct_y + pct_x + theme(legend.position=&quot;top&quot;) + 
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Germination&quot;,y=&quot;Survival&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-1.svg" width="768" /></p>
<pre class="r"><code>ggplot(germ.surv %&gt;% filter(type==&quot;mixed&quot;), aes(x=prop.alive, y=prop.hybrid, color=momsp)) + 
  geom_point() + geom_smooth(se=F, span=1.1)+ pct_y + pct_x + theme(legend.position=&quot;top&quot;) +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Survival&quot;,y=&quot;Hybrids&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-2.svg" width="768" /></p>
<pre class="r"><code>ggplot(germ.surv %&gt;% filter(type==&quot;mixed&quot;), aes(x=prop.germ, y=prop.hybrid, color=momsp)) + 
  geom_point() + geom_smooth(se=F, span=1.1)+pct_y + pct_x + theme(legend.position=&quot;top&quot;) +
  scale_color_brewer(palette=&quot;Paired&quot;) + labs(x=&quot;Germination&quot;,y=&quot;Hybrids&quot;, color=&quot;Maternal species&quot;) </code></pre>
<p><img src="figures-comp/compare-3.svg" width="768" /></p>
</div>
<div id="models" class="section level1">
<h1>Models</h1>
<p>Expand this code to see the model specifications. The models have the
following settings:</p>
<ul>
<li>response variable: proportion seeds that germinated, proportion
seedlings that survived, proportion of live seedlings that are
hybrids</li>
<li>sxc: the combined maternal plant species and crosstype (within
species, between species, mixed pollen)</li>
<li>momid: random effects of the maternal plant genotype</li>
<li>family: betabinomial (for binary variables)</li>
</ul>
<pre class="r"><code>comp.mod &lt;- list(
  prop.germ =    glmmTMB(prop.germ   ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=germ, weights = planted),
  prop.alive =   glmmTMB(prop.alive  ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=surv, weights = total),
  prop.hybrid =  glmmTMB(prop.hybrid ~ sxc + (1|momid), family=&quot;betabinomial&quot;, data=filter(surv, type==&quot;mixed&quot;), weights = alive))</code></pre>
<div id="maternal-effects" class="section level2">
<h2>Maternal effects</h2>
<pre class="r"><code>comp.ranef &lt;- map_dfr(comp.mod, ~as.data.frame(ranef(.x)), .id=&quot;trait&quot;) %&gt;% as_tibble() %&gt;% 
  rename(momid=grp, estimate = condval, sd = condsd) %&gt;% select(-component, -grpvar, -term) %&gt;% 
  mutate(momid= as.character(momid)) %&gt;% 
  left_join(count(germ, momsp.germ=momsp, momid, name=&quot;germ&quot;)) %&gt;% 
  left_join(count(surv, momsp.surv=momsp, momid, name=&quot;surv&quot;)) %&gt;% 
  mutate(momsp = coalesce(momsp.germ, momsp.surv), .keep=&quot;unused&quot;)

ggplot(comp.ranef, aes(y=estimate, ymin=estimate-sd, ymax=estimate+sd, x=paste(momsp,momid), color=momsp)) + 
  facet_wrap(vars(trait)) +  geom_pointrange() + coord_flip() + scale_color_brewer(palette=&quot;Paired&quot;)</code></pre>
<p><img src="figures-comp/ranef-1.svg" width="768" /></p>
<pre class="r"><code>comp.ranef %&gt;% select(-sd) %&gt;% pivot_wider(names_from=&quot;trait&quot;, values_from=&quot;estimate&quot;) %&gt;%
  ggplot(aes(y=prop.alive, x=prop.germ, color=momsp)) + geom_point() + scale_color_brewer(palette=&quot;Paired&quot;) +
  geom_smooth(aes(group=1), color=&quot;grey30&quot;) + theme(legend.position=&quot;top&quot;)+
  labs(x=&quot;Maternal effect on germination&quot;, y=&quot;Maternal effect on survival&quot;, color=&quot;Maternal species&quot;)</code></pre>
<p><img src="figures-comp/ranef-2.svg" width="768" /></p>
<pre class="r"><code>comp.ranef %&gt;% select(-sd) %&gt;% pivot_wider(names_from=&quot;trait&quot;, values_from=&quot;estimate&quot;) %&gt;%
  ggplot(aes(y=prop.hybrid, x=prop.alive, color=momsp)) + geom_point() + scale_color_brewer(palette=&quot;Paired&quot;) +
  geom_smooth(aes(group=1), color=&quot;grey30&quot;) + theme(legend.position=&quot;top&quot;)+
  labs(x=&quot;Maternal effect on survival&quot;, y=&quot;Maternal effect on hybrids&quot;, color=&quot;Maternal species&quot;)</code></pre>
<p><img src="figures-comp/ranef-3.svg" width="768" /></p>
</div>
<div id="inference" class="section level2">
<h2>Inference</h2>
<p>P-values for the overall effect of species and crosstype, or
collection date, on the trait.</p>
<pre class="r"><code>comp.test &lt;- map_dfr(comp.mod, ~car::Anova(.) %&gt;% tidy(), .id=&quot;trait&quot;) 
comp.test %&gt;%  mutate(term=recode(term, sxc = &quot;species and crosstype&quot;),
         p.value=format(p.value,digits=2)) %&gt;% 
  kable(digits=1, caption=&quot;ANOVA of GLMM (Type III Wald chisquare tests)&quot;)</code></pre>
<table>
<caption>ANOVA of GLMM (Type III Wald chisquare tests)</caption>
<thead>
<tr class="header">
<th align="left">trait</th>
<th align="left">term</th>
<th align="right">statistic</th>
<th align="right">df</th>
<th align="left">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">prop.germ</td>
<td align="left">species and crosstype</td>
<td align="right">277.4</td>
<td align="right">29</td>
<td align="left">2.3e-42</td>
</tr>
<tr class="even">
<td align="left">prop.alive</td>
<td align="left">species and crosstype</td>
<td align="right">353.9</td>
<td align="right">29</td>
<td align="left">1.5e-57</td>
</tr>
<tr class="odd">
<td align="left">prop.hybrid</td>
<td align="left">species and crosstype</td>
<td align="right">70.6</td>
<td align="right">9</td>
<td align="left">1.2e-11</td>
</tr>
</tbody>
</table>
<pre class="r"><code>comp.emm &lt;-  map_dfr(comp.mod, ~emmeans(., ~sxc) %&gt;% summary(type=&quot;response&quot;) %&gt;%  #average over collect date
                             tidy %&gt;% rename(estimate=prob), .id=&quot;trait&quot;) %&gt;% 
  separate(sxc, into = c(&quot;momsp&quot;,&quot;type&quot;)) %&gt;% 
  mutate(momsp=factor(momsp, levels=sp_pairs), 
         type=factor(type, levels= names(typenames)),
         trait=factor(trait, levels=names(propnames)))

#doesn&#39;t work, compares all to the first level (hookeri)
#hybrid.half &lt;- multcomp::glht(comp.mod$prop.hybrid, multcomp::mcp(sxc = paste0(sp_pairs[-7],&quot;.mixed = 0&quot;)))
#compare to null of 50% hybrids, since pollen mixtures were 50-50
hybrid.test &lt;- surv %&gt;% filter(type==&quot;mixed&quot;) %&gt;% 
  group_by(momsp, type, dadsp) %&gt;% 
  summarize(total=sum(total), alive=sum(alive), hybrid=sum(hybrid), crosses=n(), .groups=&quot;drop&quot;) %&gt;% 
  mutate(total_per_c = total/crosses, alive_per_c = alive / crosses,
         prop.hybrid = hybrid/alive,
         p = map2_dbl(hybrid, alive, ~prop.test(x=.x, n=.y, p = 0.5)$p.value),
         p.adj = p.adjust(p, &quot;bonferroni&quot;),
         stars = cut(p.adj, breaks = c(-Inf, 0.01, 0.05, 0.10, Inf), 
                labels = c(&quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;n.s.&quot;), right = FALSE))</code></pre>
</div>
</div>
<div id="plots" class="section level1">
<h1>Plots</h1>
<p>Boxplots of the germination and survival proportions in each plant x
plant cross. Germination proportions are calculated across all pots of a
given plant x plant cross. Points show estimated marginal mean and 95%
CI.</p>
<pre class="r"><code>sp_theme &lt;- theme(legend.position = &quot;top&quot;, 
                  panel.grid.major.x=element_blank(), panel.grid.minor.y=element_blank(), 
                  axis.text.x = element_text(angle=20), axis.text = element_text(color=&quot;black&quot;))
alternate &lt;- geom_tile(data=tibble(momsp=sp_pairs[seq(1,10,by=2)]), aes(x=momsp, y=0.5, height=1, width=1),  alpha=0.15, fill=&quot;grey&quot;, inherit.aes = F) 

ggplot(germ, aes(x=momsp, fill=type, y=prop.germ))  + 
  geom_boxplot(position = position_dodge(width=0.9), color=&quot;grey50&quot;, outlier.size=0.5) + alternate+ 
  geom_pointrange(data=filter(comp.emm, trait==&quot;prop.germ&quot;), aes(y=estimate, ymax=conf.high, ymin=conf.low),
                  position = position_dodge(width=0.9), size=0.3, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) +  
  labs(x=&quot;Maternal species&quot;, y=&quot;Germination&quot;, fill=&quot;Pollination&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-1.svg" width="768" /></p>
<pre class="r"><code>ggplot(surv, aes(x=momsp, fill=type, y=prop.alive))  + 
  geom_boxplot(position = position_dodge(width=0.9), color=&quot;grey50&quot;, outlier.size=0.5) + alternate+ 
  geom_pointrange(data=filter(comp.emm, trait==&quot;prop.alive&quot;), aes(y=estimate, ymax=conf.high, ymin=conf.low),
                  position = position_dodge(width=0.9), size=0.3, show.legend=F) + 
  scale_fill_manual(values=type_pal, labels=typenames) +  
  labs(x=&quot;Maternal species&quot;, y=&quot;Survival&quot;, fill=&quot;Pollination&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-2.svg" width="768" /></p>
<pre class="r"><code>ggplot(filter(surv, type==&quot;mixed&quot;), aes(x=momsp, y=prop.hybrid, fill=momsp)) + 
  geom_hline(yintercept=0.5)+ geom_violin(scale=&quot;width&quot;, draw_quantiles = c(0.25,0.5,0.75))+ 
  geom_pointrange(data=left_join(filter(comp.emm, trait==&quot;prop.hybrid&quot;), hybrid.test),
                  aes(y=estimate, ymax=conf.high, ymin=conf.low, shape=p.adj &lt; 0.05)) + 
  scale_shape_manual(&quot;Different from 50%&quot;, values=c(1, 19), labels=c(&quot;no&quot;,&quot;yes&quot;))+
  geom_text(data=hybrid.test, aes(y=1.1, label=crosses))+
  #geom_text(data=hybrid.test, aes(y=1.1, label=alive))+
  scale_fill_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) + pct_y+ 
  labs(x=&quot;Maternal species&quot;, y=&quot;Hybrids from competitive pollinations&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/rawdata-3.svg" width="768" /></p>
<p>Tested whether the proportion of hybrids significantly differed from
50%, accounting for multiple comparisons. Violin plots show proportions
for each plant x plant cross. The number of plant x plant crosses
indicated above the plot.</p>
</div>
<div id="style-length" class="section level1">
<h1>Style length</h1>
<pre class="r"><code>count(styles, momsp, plantid) %&gt;% count(momsp) %&gt;% kable(caption = &quot;Number of plants for style length&quot;)</code></pre>
<table>
<caption>Number of plants for style length</caption>
<thead>
<tr class="header">
<th align="left">momsp</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">menziesii</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">salicaria</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">lydgatei</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">sarmentosa</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">stellarioides</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">spergulina</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">membranacea</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">kauaiensis</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">kaalae</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">hookeri</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
<pre class="r"><code>styles.emm &lt;- glmmTMB(style_length_mm ~ momsp + (1|plantid), data=styles) %&gt;% 
  emmeans(~momsp) %&gt;% summary(type=&quot;response&quot;) %&gt;% tidy %&gt;% rename(style_length_mm = estimate) %&gt;% 
  left_join(mating) %&gt;% group_by(pair) %&gt;% 
  mutate(momsp=factor(momsp, levels=sp_pairs), 
         style_diff_mm = style_length_mm-min(style_length_mm))

ggplot(styles, aes(x=momsp, y=style_length_mm, fill=momsp)) + 
  geom_violin(scale=&quot;width&quot;, draw_quantiles = c(0.25,0.5,0.75))+ 
  geom_pointrange(data=styles.emm, aes(ymax=conf.high, ymin=conf.low)) + 
  scale_fill_brewer(palette=&quot;Paired&quot;, guide=&quot;none&quot;) + scale_y_continuous(limits=c(0,NA))+
  labs(x=&quot;Maternal species&quot;, y=&quot;Style length (mm)&quot;) + sp_theme</code></pre>
<p><img src="figures-comp/style_length-1.svg" width="768" /></p>
</div>
<div id="phylogenetic-distance-and-style-length" class="section level1">
<h1>Phylogenetic distance and style length</h1>
<p>Load <em>Schiedea</em> tree inferred with RAxML from Willyard et
al. nuclear dataset (no chloroplast data, 1 accession per species).</p>
<pre class="r"><code>library(ape)
library(RColorBrewer)
#devtools::install_github(&#39;gastonstat/arcdiagram&#39;)
library(igraph)
library(arcdiagram)

sp_pal &lt;- set_names(brewer.pal(length(sp_pairs), &quot;Paired&quot;), sp_pairs)
tree &lt;- read.nexus(&quot;data/RAxML_1ps_nocp_bipartitionsBranchLabels.sch.short.nobs.nex&quot;)
plot(tree, tip.color = sp_pal[tree$tip.label])</code></pre>
<p><img src="figures-comp/phylodist-1.svg" width="768" /></p>
<pre class="r"><code>par(mfrow=c(1,2))
par(mar=c(1.5,0,1.5,0))
tree_pairs &lt;- tree %&gt;% keep.tip(sp_pairs)
plot(tree_pairs, tip.color = sp_pal[tree_pairs$tip.label])
par(mar=c(0,0,0,0))
sp_pairs %&gt;% matrix(ncol=2, byrow=T) %&gt;% 
  arcplot(show.labels=F, horizontal=F, ordering=tree_pairs$tip.label, 
          col.arcs=sp_pal[(1:5)*2], col.nodes = sp_pal)</code></pre>
<p><img src="figures-comp/phylodist-2.svg" width="768" /></p>
<pre class="r"><code>par(mfrow=c(1,1))

hybrid.phylodist &lt;- comp.emm %&gt;% left_join(select(hybrid.test, momsp, dadsp, -type)) %&gt;% 
  mutate(pd = map2_dbl(as.character(momsp), as.character(dadsp), ~cophenetic(tree_pairs)[.x,.y])) %&gt;% 
  select(-df)

#TODO use a binomial GLM here, with the raw data?
hybrid.phylodist.test &lt;- hybrid.phylodist %&gt;% left_join(styles.emm, by=&quot;momsp&quot;) %&gt;% group_by(trait, type) %&gt;% nest() %&gt;% 
  mutate(model = map(data, ~lm(estimate ~ pd + style_diff_mm, data=.x)),
         anova = map(model, ~tidy(car::Anova(.x)))) %&gt;% 
  select(-data, -model) %&gt;% unnest(anova) %&gt;% filter(term !=&quot;Residuals&quot;) 

hybrid.phylodist.test %&gt;% select(trait, type, term, p.value) %&gt;% filter(type !=&quot;intra&quot;) %&gt;%
  mutate(trait = propnames[trait], type= typenames[type], 
         term=recode(term, pd=&quot;Phylogenetic distance&quot;, style_diff_mm=&quot;Style length difference&quot;)) %&gt;% 
  pivot_wider(names_from=term, values_from=p.value) %&gt;% 
  kable(caption = &quot;p-values from models of each proportion&quot;, digits=3)</code></pre>
<table>
<caption>p-values from models of each proportion</caption>
<colgroup>
<col width="15%" />
<col width="19%" />
<col width="30%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">trait</th>
<th align="left">type</th>
<th align="right">Phylogenetic distance</th>
<th align="right">Style length difference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Germinated</td>
<td align="left">Interspecific</td>
<td align="right">0.007</td>
<td align="right">0.433</td>
</tr>
<tr class="even">
<td align="left">Germinated</td>
<td align="left">Competitive</td>
<td align="right">0.044</td>
<td align="right">0.143</td>
</tr>
<tr class="odd">
<td align="left">Survived</td>
<td align="left">Interspecific</td>
<td align="right">0.307</td>
<td align="right">0.777</td>
</tr>
<tr class="even">
<td align="left">Survived</td>
<td align="left">Competitive</td>
<td align="right">0.044</td>
<td align="right">0.466</td>
</tr>
<tr class="odd">
<td align="left">Hybrid</td>
<td align="left">Competitive</td>
<td align="right">0.023</td>
<td align="right">0.141</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggplot(left_join(hybrid.phylodist, filter(hybrid.phylodist.test, term==&quot;pd&quot;)), aes(x=pd, y=estimate)) + 
  facet_grid(trait~type, labeller=as_labeller(c(propnames, typenames)))+
  geom_smooth(aes(linetype=paste(p.value &lt; 0.05, type==&quot;intra&quot;)), method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) +
  geom_pointrange(aes(ymax=conf.high, ymin=conf.low, color=momsp)) + scale_color_brewer(palette=&quot;Paired&quot;) + pct_y+ 
  scale_linetype_manual(values=c(2,0,1,0), guide=&quot;none&quot;) + scale_alpha_manual(breaks=c(T,F), values=c(0.5, 1))+
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, color=&quot;Maternal species&quot;) + theme_bw() +
  theme(axis.title.y=element_blank(), axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;,
        panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/phylodist-3.svg" width="768" /></p>
<p>Regressions against the phylogenetic distance of each species pair.
Solid regression lines are statistically significant in a model that
also included the difference in style length within pairs. Note that for
reference, data for intraspecific crosses is plotted with the
phylogenetic distance to its paired species, which doesn’t make great
sense.</p>
<p>Below, lines show the effect of style length within each pair.</p>
<pre class="r"><code>ggplot(left_join(comp.emm, select(styles.emm, momsp, style_length_mm, mating)), aes(x=style_length_mm, y=estimate)) + 
  facet_grid(trait~type, labeller=as_labeller(c(propnames, typenames)))+
  geom_line(aes(group=pair, color=momsp))+
  geom_pointrange(aes(ymax=conf.high, ymin=conf.low, color=momsp, shape=mating)) + scale_color_brewer(palette=&quot;Paired&quot;) + pct_y+ 
  labs(x=&quot;Style length (mm)&quot;, color=&quot;Maternal species&quot;, shape=&quot;Mating&quot;) + theme_bw() +
  theme(axis.title.y=element_blank(), axis.text=element_text(color=&quot;black&quot;), legend.position=&quot;top&quot;,
        panel.grid.minor = element_blank(), panel.grid.major.x=element_blank(), panel.spacing = unit(1, &quot;lines&quot;))</code></pre>
<p><img src="figures-comp/styles-1.svg" width="768" /></p>
<pre class="r"><code>hybrid.phylodist %&gt;% filter(trait==&quot;prop.hybrid&quot;,  type==&quot;mixed&quot;) %&gt;% 
  left_join(select(styles.emm, momsp, style_length_mm, mating, se=std.error)) %&gt;% 
  ggplot(aes(x=pd, y=style_length_mm, color=momsp, shape=mating)) + 
  geom_pointrange(aes(ymin=style_length_mm-se, ymax=style_length_mm+se)) + 
  scale_color_brewer(palette=&quot;Paired&quot;) + coord_cartesian(xlim=c(0, NA), ylim=c(0,NA)) +
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, y=&quot;Style length (mm)&quot;, color=&quot;Species&quot;, shape=&quot;Mating&quot;) + 
  theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figures-comp/styles-2.svg" width="768" /></p>
<p>The range of phylogenetic distances and style lengths in the study.
The species in each pair with a longer style is given a lighter
color.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
