<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="John Powers" />

<meta name="date" content="2023-05-02" />

<title>Schiedea pollen tube growth</title>

<script src="libs/header-attrs-2.20/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<script src="libs/navigation-1.1/codefolding.js"></script>
<link href="libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore"><em>Schiedea</em> pollen tube growth</h1>
<h4 class="author"><a href="http://sites.uci.edu/powers/">John
Powers</a></h4>
<h4 class="date">2023-05-02</h4>

</div>


<div id="purpose" class="section level1">
<h1>Purpose</h1>
<p>Test for post-pollination, pre-zygotic reproductive barriers between
five species pairs of <em>Schiedea</em> that have hybridization
potential, by measuring pollen tube growth from parental and hybrid
crosses at two time points.</p>
<p>Response variables:</p>
<ul>
<li>Counts - assume a Poisson distribution for count data
<ul>
<li><strong>grains</strong> - the number of pollen grains on the stigma
(only measured for some kaahoo crosses)</li>
<li><strong>mid</strong> - the number of pollen tubes counted at the
middle of the style</li>
<li><strong>bottom</strong> - the number of pollen tubes counted at the
bottom of the style</li>
</ul></li>
<li>Proportions - assume a binomial distribution for binomial data
<ul>
<li><strong>ratio</strong> - the ratio of pollen tubes at the
<strong>bottom</strong> / <strong>middle</strong> of the style, capped
at 1 because having more pollen grains at the bottom than the middle of
the style is impossible (must result from counting error). For the
binomial model, the input is the number of successes (pollen tubes at
<strong>bottom</strong>) and failures
(min(<strong>bottom</strong>/<strong>mid</strong>) -
<strong>bottom</strong>).</li>
</ul></li>
</ul>
<p>Fixed effects:</p>
<ul>
<li><strong>cross</strong> - the maternal plant x the paternal
plant</li>
<li><strong>hr</strong> - the time elapsed since pollination: 4 hr or 24
hr</li>
</ul>
<p>Random effects:</p>
<ul>
<li><strong>momid</strong> - maternal genotype, specified by its cross
and plant number</li>
<li><strong>dadid</strong> - paternal genotype, specified by its cross
and plant number</li>
<li><strong>vial</strong> - vial that holds the stigmas from one
pollination</li>
</ul>
<pre class="r"><code>library(knitr)
knitr::opts_chunk$set(comment=&quot;&quot;, cache=T, warning = F, message = F, width=999,
                      fig.path = &quot;figures/&quot;, dev = &quot;svglite&quot;, dev.args=list(fix_text_size=FALSE))
options(knitr.kable.NA = &#39;&#39;)
library(tidyverse)
library(gridExtra)
library(ggsignif) 
library(gridExtra)
library(broom)
library(emmeans) #Estimated marginal means aka least square means.
library(glmmTMB) #generalized linear mixed models with Template Model Builder

#overwrite function to make emmeans and multcomp play nice
modelparm.glmmTMB &lt;- function (model, coef. = function(x) fixef(x)[[component]],
                               vcov. = function(x) vcov(x)[[component]],
                               df = NULL, component=&quot;cond&quot;, ...) {
    multcomp:::modelparm.default(model, coef. = coef., vcov. = vcov.,
                        df = df, ...)
}

signif.num &lt;- function(x) {
   cut(x, breaks = c(0, 0.001, 0.01, 0.05, 1), 
           labels = c(&quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot; &quot;), include.lowest = T)
}</code></pre>
</div>
<div id="read-data" class="section level1">
<h1>Read data</h1>
<pre class="r"><code>#long format: salmen (1 stigma/vial), lydsar (2 stigmas/vial), kaahoo (3 stigmas/vial without letter)
#wide format: memkau (3-5 stigmas/vial), speste (6 stigmas/vial), kaahoo_extra(3-5 stigmas/vial)
pops &lt;- list(sali=&quot;842&quot;, menz = c(&quot;849&quot;, &quot;950&quot;), 
             lydg=&quot;6682&quot;, sarm=&quot;896&quot;, 
             sper= c(&quot;120289&quot;,&quot;120294&quot;,&quot;120296&quot;), stel = c(&quot;14651&quot;,&quot;14765&quot;),
             memb = c(&quot;mem&quot;,&quot;864&quot;), kaua = &quot;kaua&quot;,
               kaal = c(&quot;892&quot;, &quot;3587&quot;), hook = c(&quot;794&quot;, &quot;866&quot;, &quot;879&quot;,&quot;899&quot;))

pair_labels &lt;- c(salmen = &quot;S. salicaria | S. menziesii&quot;,
                 lydsar = &quot;S. lydgatei | S. sarmentosa&quot;,
                 speste = &quot;S. spergulina | S. stellariodes&quot;,
                 memkau = &quot;S. membranacea | S. kauaiensis&quot;,
                 kaahoo = &quot;S. kaalae | S. hookeri&quot;)

abbrev &lt;- list(salmen = c(S=&quot;Sl&quot;,M=&quot;Mn&quot;), 
               lydsar = c(L=&quot;Ly&quot;,S=&quot;Sr&quot;), 
               speste = c(Sp=&quot;Sp&quot;,St=&quot;St&quot;), 
               memkau = c(M=&quot;Mb&quot;,K=&quot;Ki&quot;),
               kaahoo = c(K=&quot;Kl&quot;,H=&quot;Ho&quot;))

species_names &lt;- c(&quot;Mn&quot;=&quot;menziesii&quot;, &quot;Sl&quot;=&quot;salicaria&quot;,
              &quot;Ly&quot;=&quot;lydgatei&quot;, &quot;Sr&quot; =&quot;sarmentosa&quot;,
              &quot;St&quot;=&quot;stellarioides&quot;,&quot;Sp&quot;=&quot;spergulina&quot;,
              &quot;Mb&quot;=&quot;membranacea&quot;,&quot;Ki&quot;=&quot;kauaiensis&quot;,
              &quot;Kl&quot;=&quot;kaalae&quot;, &quot;Ho&quot;=&quot;hookeri&quot;)

cross_levels &lt;- map(abbrev, ~ c(paste0(.x[1],&quot;x&quot;,.x[1]), paste0(.x[1],&quot;x&quot;,.x[2]), paste0(.x[2],&quot;x&quot;,.x[2]), paste0(.x[2],&quot;x&quot;,.x[1])))

tubesl &lt;- list()             
# kaalae - hookeri
tubesl[[&quot;kaahoo&quot;]] &lt;- read_csv(&quot;data/pollentubes_kaahoo_extra.csv&quot;) %&gt;% 
  pivot_longer(starts_with(c(&quot;mid&quot;,&quot;bottom&quot;)), names_to=c(&quot;location&quot;,&quot;stigma&quot;), names_sep=&quot;_&quot;, values_to=&quot;n_tubes&quot;) %&gt;% 
  pivot_wider(names_from=&quot;location&quot;, values_from=&quot;n_tubes&quot;) %&gt;% 
  drop_na(mid, bottom) %&gt;% #one entry has mid but no bottom count
  mutate(momcross = toupper(str_sub(momsp,1,1)), dadcross = toupper(str_sub(dadsp,1,1)),
         observer=&quot;SGW&quot;) %&gt;% 
  bind_rows(read_csv(&quot;data/pollentubes_kaahoo.csv&quot;) %&gt;% #edited from pollentubes.csv
              separate(cross, into=c(&quot;momcross&quot;,&quot;dadcross&quot;), sep=&quot;x&quot;) %&gt;% 
              mutate(observer=&quot;SZ&quot;,
                     vial = as.numeric(str_remove(vial, &quot;[ABCD]&quot;)))) %&gt;% 
  mutate(momid = paste(momcross, momid), dadid = paste(dadcross, dadid))

# lydgatei- - sarmentosa
tubesl[[&quot;lydsar&quot;]] &lt;- read_csv(&quot;data/pollentubes_lydsar.csv&quot;) %&gt;% 
  filter(keep == 1, hr != 2) %&gt;% select(-keep) %&gt;%  #exclude the resamples of pollinations that gave no tubes, and counts at 2 hrs
  mutate(mompop = as.character(mompop), dadpop=as.character(dadpop),
         momid = paste(mompop, momplant), dadid = paste(dadpop, dadplant),
         momcross = ifelse(mompop %in% pops[[&quot;sarm&quot;]], &quot;S&quot;, &quot;L&quot;), dadcross = ifelse(dadpop %in% pops[[&quot;sarm&quot;]], &quot;S&quot;, &quot;L&quot;))

#salicaria - menziesii
tubesl[[&quot;salmen&quot;]] &lt;- read_csv(&quot;data/pollentubes_salmen.csv&quot;) %&gt;% 
  select(-cross) %&gt;% 
  mutate(mompop = str_sub(momid, 1, 3), dadpop = str_sub(dadid, 1, 3),
         momcross = ifelse(mompop %in% pops[[&quot;sali&quot;]], &quot;S&quot;, &quot;M&quot;), dadcross = ifelse(dadpop %in% pops[[&quot;sali&quot;]], &quot;S&quot;, &quot;M&quot;))

# membranacea - kauaiensis
tubesl[[&quot;memkau&quot;]] &lt;- read_csv(&quot;data/pollentubes_memkau.csv&quot;) %&gt;% mutate(across(ends_with(&quot;sp&quot;), as.factor)) %&gt;% 
  pivot_longer(starts_with(c(&quot;mid&quot;,&quot;bottom&quot;)), names_to=c(&quot;location&quot;,&quot;stigma&quot;), names_sep=&quot;_&quot;, values_to=&quot;n_tubes&quot;) %&gt;% 
  pivot_wider(names_from=&quot;location&quot;, values_from=&quot;n_tubes&quot;) %&gt;% 
  mutate(status = fct_explicit_na(status, &quot;OK&quot;),
         mid =    ifelse(status==&quot;no pollen on stigmas&quot;, NA, mid), #set to NA to drop, 0 to keep
         bottom = ifelse(status==&quot;no pollen on stigmas&quot;, NA, bottom)) %&gt;% #see SGW email 2022-05-09, NOTES tab
  drop_na(mid) %&gt;% 
  mutate(momcross = ifelse(momsp %in% pops[[&quot;memb&quot;]], &quot;M&quot;, &quot;K&quot;),  dadcross = ifelse(dadsp %in% pops[[&quot;memb&quot;]], &quot;M&quot;, &quot;K&quot;),
         momid = paste(momcross, momplant), dadid = paste(dadcross, dadplant),)

#spergulina - stellarioides
tubesl[[&quot;speste&quot;]] &lt;- read_csv(&quot;data/pollentubes_speste.csv&quot;) %&gt;% mutate(across(ends_with(&quot;pop&quot;), as.factor)) %&gt;% 
  pivot_longer(starts_with(c(&quot;mid&quot;,&quot;bottom&quot;)), names_to=c(&quot;location&quot;,&quot;stigma&quot;), names_sep=&quot;_&quot;, values_to=&quot;n_tubes&quot;) %&gt;% 
  pivot_wider(names_from=&quot;location&quot;, values_from=&quot;n_tubes&quot;) %&gt;% drop_na(mid) %&gt;% 
  mutate(momid = paste(mompop, momplant), dadid = paste(dadpop, dadplant),
         momcross = ifelse(mompop %in% pops[[&quot;sper&quot;]], &quot;Sp&quot;, &quot;St&quot;), dadcross = ifelse(dadpop %in% pops[[&quot;sper&quot;]], &quot;Sp&quot;, &quot;St&quot;))

tubes &lt;- bind_rows(tubesl, .id = &quot;pair&quot;) %&gt;% 
  mutate(hr = factor(paste0(hr,&quot;hr&quot;), levels=c(&quot;4hr&quot;,&quot;24hr&quot;)),
         ratioraw = bottom/mid,
         adjmid = ifelse(bottom &gt; mid, bottom, mid),
         ratio = bottom/adjmid,
         justmid = mid - bottom,
         vial = as.character(vial), #for random effect of pollination
         momcross = map2_chr(pair, momcross, ~ abbrev[[.x]][[.y]]),
         dadcross = map2_chr(pair, dadcross, ~ abbrev[[.x]][[.y]]),
         cross = factor(paste0(momcross,&quot;x&quot;,dadcross), levels=unlist(cross_levels)),
         crosshr = factor(paste0(cross, hr), levels = paste(rep(levels(cross),each=2), levels(hr), sep=&quot;&quot;)),
         pair = factor(pair, levels=names(pair_labels))) %&gt;% arrange(pair)
str(tubes)</code></pre>
<pre><code>tibble [1,943 × 31] (S3: tbl_df/tbl/data.frame)
 $ pair             : Factor w/ 5 levels &quot;salmen&quot;,&quot;lydsar&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...
 $ line             : num [1:1943] 114 422 423 454 455 330 331 452 453 206 ...
 $ emasculation_date: Date[1:1943], format: NA NA ...
 $ pollination_date : Date[1:1943], format: &quot;2020-01-08&quot; &quot;2020-01-17&quot; ...
 $ momsp            : chr [1:1943] NA NA NA NA ...
 $ momid            : chr [1:1943] &quot;842 201-46&quot; &quot;842 201-46&quot; &quot;842 201-46&quot; &quot;842 201-46&quot; ...
 $ dadsp            : chr [1:1943] NA NA NA NA ...
 $ dadid            : chr [1:1943] &quot;842 221F&quot; &quot;842 221F&quot; &quot;842 221F&quot; &quot;842 51-20&quot; ...
 $ vial             : chr [1:1943] &quot;83&quot; &quot;408&quot; &quot;409&quot; &quot;442&quot; ...
 $ hr               : Factor w/ 2 levels &quot;4hr&quot;,&quot;24hr&quot;: 1 1 2 1 2 1 2 1 2 1 ...
 $ count_date       : Date[1:1943], format: NA NA ...
 $ notes            : chr [1:1943] NA NA NA NA ...
 $ stigma           : chr [1:1943] NA NA NA NA ...
 $ mid              : num [1:1943] 9 4 9 8 13 6 9 9 12 4 ...
 $ bottom           : num [1:1943] 3 1 7 3 11 1 9 3 11 1 ...
 $ momcross         : chr [1:1943] &quot;Sl&quot; &quot;Sl&quot; &quot;Sl&quot; &quot;Sl&quot; ...
 $ dadcross         : chr [1:1943] &quot;Sl&quot; &quot;Sl&quot; &quot;Sl&quot; &quot;Sl&quot; ...
 $ observer         : chr [1:1943] NA NA NA NA ...
 $ grains           : num [1:1943] NA NA NA NA NA NA NA NA NA NA ...
 $ mompop           : chr [1:1943] &quot;842&quot; &quot;842&quot; &quot;842&quot; &quot;842&quot; ...
 $ momplant         : chr [1:1943] NA NA NA NA ...
 $ dadpop           : chr [1:1943] &quot;842&quot; &quot;842&quot; &quot;842&quot; &quot;842&quot; ...
 $ dadplant         : chr [1:1943] NA NA NA NA ...
 $ collect_date     : Date[1:1943], format: &quot;2020-01-08&quot; NA ...
 $ status           : Factor w/ 3 levels &quot;no pollen on stigmas&quot;,..: NA NA NA NA NA NA NA NA NA NA ...
 $ ratioraw         : num [1:1943] 0.333 0.25 0.778 0.375 0.846 ...
 $ adjmid           : num [1:1943] 9 4 9 8 13 6 9 9 12 4 ...
 $ ratio            : num [1:1943] 0.333 0.25 0.778 0.375 0.846 ...
 $ justmid          : num [1:1943] 6 3 2 5 2 5 0 6 1 3 ...
 $ cross            : Factor w/ 20 levels &quot;SlxSl&quot;,&quot;SlxMn&quot;,..: 1 1 1 1 1 2 2 2 2 1 ...
 $ crosshr          : Factor w/ 40 levels &quot;SlxSl4hr&quot;,&quot;SlxSl24hr&quot;,..: 1 1 2 1 2 3 4 3 4 1 ...</code></pre>
</div>
<div id="inventory-of-crosses" class="section level1">
<h1>Inventory of crosses</h1>
<pre class="r"><code>kable(with(tubes, table(cross, hr)), caption=&quot;Number of styles of each cross type&quot;)</code></pre>
<table>
<caption>Number of styles of each cross type</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">4hr</th>
<th align="right">24hr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SlxSl</td>
<td align="right">17</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">SlxMn</td>
<td align="right">18</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">MnxMn</td>
<td align="right">22</td>
<td align="right">22</td>
</tr>
<tr class="even">
<td align="left">MnxSl</td>
<td align="right">20</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">LyxLy</td>
<td align="right">28</td>
<td align="right">28</td>
</tr>
<tr class="even">
<td align="left">LyxSr</td>
<td align="right">28</td>
<td align="right">28</td>
</tr>
<tr class="odd">
<td align="left">SrxSr</td>
<td align="right">16</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">SrxLy</td>
<td align="right">18</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">SpxSp</td>
<td align="right">52</td>
<td align="right">53</td>
</tr>
<tr class="even">
<td align="left">SpxSt</td>
<td align="right">66</td>
<td align="right">64</td>
</tr>
<tr class="odd">
<td align="left">StxSt</td>
<td align="right">78</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="left">StxSp</td>
<td align="right">42</td>
<td align="right">42</td>
</tr>
<tr class="odd">
<td align="left">MbxMb</td>
<td align="right">75</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">MbxKi</td>
<td align="right">87</td>
<td align="right">95</td>
</tr>
<tr class="odd">
<td align="left">KixKi</td>
<td align="right">73</td>
<td align="right">76</td>
</tr>
<tr class="even">
<td align="left">KixMb</td>
<td align="right">67</td>
<td align="right">68</td>
</tr>
<tr class="odd">
<td align="left">KlxKl</td>
<td align="right">72</td>
<td align="right">55</td>
</tr>
<tr class="even">
<td align="left">KlxHo</td>
<td align="right">53</td>
<td align="right">38</td>
</tr>
<tr class="odd">
<td align="left">HoxHo</td>
<td align="right">97</td>
<td align="right">79</td>
</tr>
<tr class="even">
<td align="left">HoxKl</td>
<td align="right">66</td>
<td align="right">55</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(with(tubes, table(pair, hr)), caption=&quot;Number of styles of each species pair&quot;)</code></pre>
<table>
<caption>Number of styles of each species pair</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">4hr</th>
<th align="right">24hr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">salmen</td>
<td align="right">77</td>
<td align="right">76</td>
</tr>
<tr class="even">
<td align="left">lydsar</td>
<td align="right">90</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">speste</td>
<td align="right">238</td>
<td align="right">236</td>
</tr>
<tr class="even">
<td align="left">memkau</td>
<td align="right">302</td>
<td align="right">319</td>
</tr>
<tr class="odd">
<td align="left">kaahoo</td>
<td align="right">288</td>
<td align="right">227</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(tubes %&gt;% count(pair, cross, vial) %&gt;% with(table(pair, n)), caption = &quot;Number of styles counted per vial&quot;)</code></pre>
<table>
<caption>Number of styles counted per vial</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
<th align="right">6</th>
<th align="right">8</th>
<th align="right">9</th>
<th align="right">10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">salmen</td>
<td align="right">153</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">lydsar</td>
<td align="right">0</td>
<td align="right">90</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">speste</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">74</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">memkau</td>
<td align="right">5</td>
<td align="right">11</td>
<td align="right">91</td>
<td align="right">37</td>
<td align="right">24</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">kaahoo</td>
<td align="right">5</td>
<td align="right">19</td>
<td align="right">121</td>
<td align="right">19</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="models" class="section level1">
<h1>Models</h1>
<pre class="r"><code>pair_contrasts &lt;- function(mod) {
  lv &lt;- levels(mod$frame$crosshr)
  crosshr.x &lt;- set_names(rep(1:4, each=2) + rep(c(-0.25, 0.25), 4), lv)
  tube.contrasts &lt;- list(c(5,6), c(7,8), c(5,7), c(6,8),
                         c(3,4), c(1,2), c(3,1), c(4,2), 
                         c(5,1), c(6,2)) %&gt;% 
    map_chr(~paste(lv[.x[1]], &quot;-&quot;, lv[.x[2]], &quot;== 0&quot;))
  tube.contrasts &lt;- tube.contrasts[c(3,4,7,8)] #TODO exclude time and between-species comparisons for plotting
  
  multcomp::glht(mod, linfct = multcomp::mcp(crosshr=tube.contrasts)) %&gt;%
  multcomp:::summary.glht(test=multcomp::adjusted(type=&quot;fdr&quot;)) %&gt;% tidy() %&gt;% 
  separate(contrast, c(&quot;lvl1&quot;, &quot;lvl2&quot;), sep = &quot; - &quot;, extra=&quot;merge&quot;) %&gt;%
  mutate(signif = adj.p.value &lt; 0.05, 
         stars = signif.num(adj.p.value), 
         height =  c(rep(c(1,1,2,3),2),4,5)[c(3,4,7,8)], 
         x1 = crosshr.x[lvl1], 
         x2 = crosshr.x[lvl2])
}

pair_emm &lt;- function(mod) {
  emm &lt;- ref_grid(mod) %&gt;% emmeans(~ crosshr) %&gt;% summary() %&gt;% 
    mutate(cross = factor(str_sub(crosshr,1,5),  levels=unlist(cross_levels)),
           hr =    factor(str_sub(crosshr,6,10), levels=c(&quot;4hr&quot;,&quot;24hr&quot;)), .keep=&quot;unused&quot;)
  if(mod$modelInfo$family$link == &quot;logit&quot;) {
    emm %&gt;% mutate(response = plogis(emmean), uSE = plogis(emmean+SE), lSE = plogis(emmean-SE))
  } else if(mod$modelInfo$family$link == &quot;log&quot;) {
    emm %&gt;% mutate(response = exp(emmean), uSE = exp(emmean+SE), lSE = exp(emmean-SE))
  }
}

tubes.mod &lt;- tubes %&gt;% group_by(pair) %&gt;% nest() %&gt;% #TODO got rid of 1|dadid due to extremely small estimates in some species, OK?
  mutate(mid_mod   = map(data, ~ glmmTMB(mid~crosshr +  (1|momid) + (1|vial), data=.x, family=&quot;poisson&quot;)),
         mid_emm    = map(mid_mod, pair_emm),
         mid_cont   = map(mid_mod, pair_contrasts),
         bottom_mod= map(data, ~ glmmTMB(bottom~crosshr  + (1|momid) + (1|vial), data=.x, family=&quot;poisson&quot;)),
         bottom_emm = map(bottom_mod, pair_emm),
         bottom_cont= map(bottom_mod, pair_contrasts),
         ratio_mod = map(data, ~ glmmTMB(cbind(bottom, adjmid-bottom)~crosshr  + (1|momid) + (1|vial), data=.x, family=&quot;binomial&quot;)),
         ratio_emm  = map(ratio_mod, pair_emm),
         ratio_cont = map(ratio_mod, pair_contrasts))

#diagnose random effects
# library(broom.mixed)
# tubes.mod %&gt;% mutate(ranef_midmod2 = map(bottom_mod, tidy)) %&gt;% select(ranef_midmod2) %&gt;% unnest(ranef_midmod2) %&gt;% 
#   filter(effect == &quot;ran_pars&quot;) %&gt;% 
#   ggplot(aes(x=pair, color=group, y=log10(estimate))) + geom_point()</code></pre>
<pre class="r"><code>mytheme &lt;- theme(legend.text=element_text(size=rel(1)), legend.position=&quot;right&quot;, 
                 axis.text = element_text(colour=&quot;black&quot;, size=rel(1)), text=element_text(size=14), 
                 axis.ticks.x = element_blank(), plot.title = element_text(face=&quot;italic&quot;, size=14))
timefills &lt;-   scale_fill_brewer(palette=3, type=&quot;qual&quot;, labels=c(&quot;4 hr&quot;,&quot;24 hr&quot;))

var_labels &lt;- c(mid = &quot;Pollen tubes at middle of style&quot;,
                bottom = &quot;Pollen tubes at bottom of style&quot;,
                ratio = &quot;Ratio of pollen tubes at bottom : middle of style&quot;,
                justmid = &quot;Difference of pollen tubes at middle and bottom of style&quot;)

plot_all &lt;- function(yvar) {
ggplot(unnest(select(tubes.mod, paste0(yvar,&quot;_emm&quot;)), paste0(yvar,&quot;_emm&quot;))) +
  facet_wrap(vars(pair), scales=&quot;free_x&quot;, nrow=1, labeller = as_labeller(pair_labels)) +
  geom_violin(data=unnest(select(tubes.mod, data), data),  aes(y=.data[[yvar]], x=cross, fill=hr),position=position_dodge(), 
              draw_quantiles=c(0.25,0.5,0.75), linewidth=0.5, bw=ifelse(yvar==&quot;ratio&quot;,0.1,1)) +
  geom_pointrange(aes(y=response, x=cross, fill=hr, ymin=lSE, ymax=uSE), position=position_dodge2(0.9)) +
  geom_signif(data=unnest(select(tubes.mod, paste0(yvar,&quot;_cont&quot;)), paste0(yvar,&quot;_cont&quot;)) %&gt;% 
                mutate(n=row_number(), stars=replace_na(as.character(stars), &quot;NA&quot;), #TODO speste contrasts are failing, fitting problem gives NAs here
                       hr = fct_recode(factor(height), &quot;4hr&quot;=&quot;2&quot;, &quot;24hr&quot;=&quot;3&quot;),
                       ypos=(height-1)/c(mid=1, bottom=1, ratio=15)[yvar]+c(mid=20, bottom=20, ratio=1)[yvar]),
              aes(y_position=ypos, xmin = x1, xmax = x2, annotations=stars, group=n), # color=hr #TODO signif color not working
              vjust=0.5, tip_length=0.01, manual=T) +
  labs(x=&quot;&quot;, y=var_labels[yvar]) +
  scale_y_continuous(limits=c(0,c(mid=22, bottom=22, ratio=17/15)[yvar]), 
                     expand = expansion(add=c(0,c(mid=0.5, bottom=0.5, ratio=0.05)[yvar])), 
                     breaks = seq(0,1,by=0.25) * c(mid = 20, bottom = 20, ratio = 1)[yvar])+
  scale_x_discrete(labels=~str_replace(.x, &quot;x&quot;,&quot; x\n&quot;)) + 
  theme_classic() + mytheme + timefills + guides(fill=&quot;none&quot;)
}
plot_all(&quot;mid&quot;)</code></pre>
<p><img src="figures/pollentubes_all-1.svg" width="1248" /></p>
<pre class="r"><code>plot_all(&quot;bottom&quot;)</code></pre>
<p><img src="figures/pollentubes_all-2.svg" width="1248" /></p>
<pre class="r"><code>plot_all(&quot;ratio&quot;)</code></pre>
<p><img src="figures/pollentubes_all-3.svg" width="1248" /></p>
<pre class="r"><code>plot_emm &lt;- function(pair, emm, cont, data, yvar) {
  ggplot(emm, aes(y=response, x=cross, fill=hr)) +
    geom_violin(data=data, aes_string(y=yvar),position=position_dodge(),
                draw_quantiles=c(0.25,0.5,0.75), linewidth=0.5, bw=ifelse(yvar==&quot;ratio&quot;,0.1,1)) +
    geom_pointrange(aes(ymin=lSE, ymax=uSE), position=position_dodge2(0.9)) +
    geom_signif(data=emm, y_position=cont$height/c(mid=1, bottom=1, ratio=15)[yvar]+c(mid=19, bottom=19, ratio=1)[yvar],
                xmin = cont$x1, xmax = cont$x2, annotations=replace_na(as.character(cont$stars), &quot;NA&quot;), vjust=0.5, tip_length=0.01) +
    labs(x=&quot;&quot;, y=var_labels[yvar], title=pair_labels[pair]) +
    scale_y_continuous(limits=c(0,c(mid=24, bottom=24, ratio=1.4)[yvar]),
                       expand = expansion(add=c(0,c(mid=0.5, bottom=0.5, ratio=0.05)[yvar])),
                       breaks = seq(0,1,by=0.25) * c(mid = 20, bottom = 20, ratio = 1)[yvar])+
    scale_x_discrete(labels=~str_replace(.x, &quot;x&quot;,&quot; x &quot;)) +
    theme_classic() + mytheme + timefills + guides(fill=&quot;none&quot;)
}

grid.arrange(grobs=pmap(select(tubes.mod, pair, emm=mid_emm, cont=mid_cont, data),
                        plot_emm, yvar=&quot;mid&quot;), nrow=1)
grid.arrange(grobs=pmap(select(tubes.mod, pair, emm=bottom_emm, cont=bottom_cont, data),
                        plot_emm, yvar=&quot;bottom&quot;), nrow=1)
grid.arrange(grobs=pmap(select(tubes.mod, pair, emm=ratio_emm, cont=ratio_cont, data),
                        plot_emm, yvar=&quot;ratio&quot;), nrow=1)</code></pre>
</div>
<div id="tubes-at-bottom-v.-middle-of-style" class="section level1">
<h1>Tubes at bottom v. middle of style</h1>
<pre class="r"><code>ggplot(tubes, aes(x=bottom, y=mid)) + 
  facet_wrap(vars(pair), labeller=as_labeller(pair_labels))+
  labs(x=var_labels[&quot;bottom&quot;], y=var_labels[&quot;mid&quot;]) +
  geom_abline(slope=1, size=1) +
  geom_point(alpha=0.1) + 
  coord_fixed(xlim=c(0,NA),ylim=c(0,NA))+
  theme_classic()+mytheme+theme(strip.text=element_text(face=&quot;italic&quot;))</code></pre>
<p><img src="figures/pollentubes_diff-1.svg" width="864" /></p>
</div>
<div id="phylogenetic-distance" class="section level1">
<h1>Phylogenetic distance</h1>
<pre class="r"><code>library(ape)
tree &lt;- read.nexus(&quot;data/RAxML_1ps_nocp_bipartitionsBranchLabels.sch.short.nobs.nex&quot;)

tubes.phylodist &lt;- tubes.mod %&gt;% select(pair, ends_with(&quot;_emm&quot;)) %&gt;% pivot_longer(ends_with(&quot;_emm&quot;), names_to=&quot;trait&quot;) %&gt;% 
  mutate(trait = str_remove(trait, &quot;_emm&quot;)) %&gt;% unnest(value) %&gt;% 
  mutate(momsp = factor(species_names[str_sub(cross, 1,2)], levels=species_names)) %&gt;% group_by(pair, trait, momsp, hr) %&gt;% 
  summarize(trait_diff = response[1]-response[2], 
            dadsp = species_names[str_sub(cross[2], 4,5)], .groups = &quot;drop&quot;) %&gt;% 
  mutate(pd = map2_dbl(as.character(momsp), as.character(dadsp), ~cophenetic(tree)[.x,.y]))

ggplot(tubes.phylodist, aes(x=pd, y=trait_diff)) + facet_grid(trait~hr, scales=&quot;free_y&quot;) +
  geom_point(aes(color=momsp), size=2) + geom_smooth(method=&quot;lm&quot;, se=F, color=&quot;grey&quot;) + 
  scale_color_brewer(palette=&quot;Paired&quot;)+
  labs(x=&quot;Phylogenetic distance (mutations/site)&quot;, color=&quot;Maternal species&quot;,
       y=&quot;Difference between intra and interspecies cross&quot;) + theme_bw() + 
  theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figures/phylo_tubes-1.svg" width="672" /></p>
</div>
<div id="package-citations" class="section level1">
<h1>Package citations</h1>
<pre class="r"><code>citation(&quot;glmmTMB&quot;)</code></pre>
<pre><code>To cite glmmTMB in publications use:

  Mollie E. Brooks, Kasper Kristensen, Koen J. van Benthem, Arni
  Magnusson, Casper W. Berg, Anders Nielsen, Hans J. Skaug, Martin
  Maechler and Benjamin M. Bolker (2017). glmmTMB Balances Speed and
  Flexibility Among Packages for Zero-inflated Generalized Linear Mixed
  Modeling. The R Journal, 9(2), 378-400. doi: 10.32614/RJ-2017-066.

A BibTeX entry for LaTeX users is

  @Article{,
    author = {Mollie E. Brooks and Kasper Kristensen and Koen J. {van Benthem} and Arni Magnusson and Casper W. Berg and Anders Nielsen and Hans J. Skaug and Martin Maechler and Benjamin M. Bolker},
    title = {{glmmTMB} Balances Speed and Flexibility Among Packages for Zero-inflated Generalized Linear Mixed Modeling},
    year = {2017},
    journal = {The R Journal},
    doi = {10.32614/RJ-2017-066},
    pages = {378--400},
    volume = {9},
    number = {2},
  }</code></pre>
<pre class="r"><code>citation(&quot;emmeans&quot;)</code></pre>
<pre><code>To cite package &#39;emmeans&#39; in publications use:

  Lenth R (2023). _emmeans: Estimated Marginal Means, aka Least-Squares
  Means_. R package version 1.8.5,
  &lt;https://CRAN.R-project.org/package=emmeans&gt;.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {emmeans: Estimated Marginal Means, aka Least-Squares Means},
    author = {Russell V. Lenth},
    year = {2023},
    note = {R package version 1.8.5},
    url = {https://CRAN.R-project.org/package=emmeans},
  }</code></pre>
<pre class="r"><code>citation(&quot;multcomp&quot;)</code></pre>
<pre><code>Please cite the multcomp package by the following reference:

  Hothorn T, Bretz F, Westfall P (2008). &quot;Simultaneous Inference in
  General Parametric Models.&quot; _Biometrical Journal_, *50*(3), 346-363.

A BibTeX entry for LaTeX users is

  @Article{,
    title = {Simultaneous Inference in General Parametric Models},
    author = {Torsten Hothorn and Frank Bretz and Peter Westfall},
    journal = {Biometrical Journal},
    year = {2008},
    volume = {50},
    number = {3},
    pages = {346--363},
  }</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
